--- /Users/haoye.tian/Documents/University/data/PatchCollecting/jKali/Incorrect/Chart/patch1-Chart-5-jKali.buggy	2020-11-02 19:13:13.000000000 +0100
+++ /Users/haoye.tian/Documents/University/data/PatchCollecting/jKali/Incorrect/Chart/patch1-Chart-5-jKali.fixed	2020-11-02 19:13:13.000000000 +0100
@@ -557,13 +557,14 @@
         }
         else {
             // if the series is sorted, the negative index is a result from
-            // Collections.binarySearch() and tells us where to insert the
-            // new item...otherwise it will be just -1 and we should just
-            // append the value to the list...
-            if (this.autoSort) {
-                this.data.add(-index - 1, new XYDataItem(x, y));
-            }
-            else {
+ 
+ 
+ 
+			if (false) {
+ 				this.data.add(((-index) - 1), new XYDataItem(x, y));
+ 			}else 
+ 			{
+
                 this.data.add(new XYDataItem(x, y));
             }
             // check if this addition will exceed the maximum item count...
