--- MultiplePiePlot.java	2020-10-11 16:12:28.000000000 +0200
+++ MultiplePiePlot copy.java	2020-11-05 21:01:47.000000000 +0100
@@ -142,7 +142,7 @@
      */
     public MultiplePiePlot(CategoryDataset dataset) {
         super();
-        this.dataset = dataset;
+        this.setDataset( dataset ) ;
         PiePlot piePlot = new PiePlot(null);
         this.pieChart = new JFreeChart(piePlot);
         this.pieChart.removeLegend();
