--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_112/TypeInference/Closure_112_TypeInference_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_112/TypeInference/Closure_112_TypeInference_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -1189,9 +1189,15 @@
     }
 
     // Try to infer the template types
-    Map<TemplateType, JSType> inferred = 
-        inferTemplateTypesFromParameters(fnType, n);
-
+    Map<TemplateType, JSType> inferred = Maps.filterKeys(
+        inferTemplateTypesFromParameters(fnType, n),
+        new Predicate<TemplateType>() {
+
+          @Override
+          public boolean apply(TemplateType key) {
+            return keys.contains(key);
+          }}
+        );
 
     // Replace all template types. If we couldn't find a replacement, we
     // replace it with UNKNOWN.
