--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_104/Gamma/Math_104_Gamma_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_104/Gamma/Math_104_Gamma_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -34,7 +34,7 @@
     private static final long serialVersionUID = -6587513359895466954L;
 
     /** Maximum allowed numerical error. */
-    private static final double DEFAULT_EPSILON = 10e-9;
+    private static final double DEFAULT_EPSILON = 10e-15;
 
     /** Lanczos coefficients */
     private static double[] lanczos =
