--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_10/DSCompiler/Math_10_DSCompiler_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_10/DSCompiler/Math_10_DSCompiler_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -1416,6 +1416,7 @@
         }
 
         // fix value to take special cases (+0/+0, +0/-0, -0/+0, -0/-0, +/-infinity) correctly
+        result[resultOffset] = FastMath.atan2(y[yOffset], x[xOffset]);
 
     }
 
