--- /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch75/Math_95/FDistributionImpl/Math_95_FDistributionImpl_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch75/Math_95/FDistributionImpl/Math_95_FDistributionImpl_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -145,7 +145,7 @@
         double d = getDenominatorDegreesOfFreedom();
             // use mean
             ret = d / (d - 2.0);
-        return ret;
+        return numeratorDegreesOfFreedom;
     }
     
     /**
