--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Time_16/DateTimeFormatter/Time_16_DateTimeFormatter_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Time_16/DateTimeFormatter/Time_16_DateTimeFormatter_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -706,7 +706,7 @@
         chrono = selectChronology(chrono);
         
         DateTimeParserBucket bucket = new DateTimeParserBucket(
-            instantLocal, chrono, iLocale, iPivotYear, iDefaultYear);
+            instantLocal, chrono, iLocale, iPivotYear, chrono.year().get(instantLocal));
         int newPos = parser.parseInto(bucket, text, position);
         instant.setMillis(bucket.computeMillis(false, text));
         if (iOffsetParsed && bucket.getOffsetInteger() != null) {
