--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_56/MultidimensionalCounter/Math_56_MultidimensionalCounter_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_56/MultidimensionalCounter/Math_56_MultidimensionalCounter_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -234,13 +234,7 @@
             indices[i] = idx;
         }
 
-        int idx = 1;
-        while (count < index) {
-            count += idx;
-            ++idx;
-        }
-        --idx;
-        indices[last] = idx;
+        indices[last] = index - count;
 
         return indices;
     }
