--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_39/EmbeddedRungeKuttaIntegrator/Math_39_EmbeddedRungeKuttaIntegrator_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_39/EmbeddedRungeKuttaIntegrator/Math_39_EmbeddedRungeKuttaIntegrator_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -247,6 +247,15 @@
         }
 
         stepSize = hNew;
+        if (forward) {
+            if (stepStart + stepSize >= t) {
+                stepSize = t - stepStart;
+            }
+        } else {
+            if (stepStart + stepSize <= t) {
+                stepSize = t - stepStart;
+            }
+        }
 
         // next stages
         for (int k = 1; k < stages; ++k) {
