--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_78/PeepholeFoldConstants/Closure_78_PeepholeFoldConstants_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_78/PeepholeFoldConstants/Closure_78_PeepholeFoldConstants_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -708,14 +708,12 @@
         break;
       case Token.MOD:
         if (rval == 0) {
-          error(DiagnosticType.error("JSC_DIVIDE_BY_0_ERROR", "Divide by 0"), right);
           return null;
         }
         result = lval % rval;
         break;
       case Token.DIV:
         if (rval == 0) {
-          error(DiagnosticType.error("JSC_DIVIDE_BY_0_ERROR", "Divide by 0"), right);
           return null;
         }
         result = lval / rval;
