--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_3/MathArrays/Math_3_MathArrays_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_3/MathArrays/Math_3_MathArrays_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -818,7 +818,10 @@
             throw new DimensionMismatchException(len, b.length);
         }
 
+        if (len == 1) {
             // Revert to scalar multiplication.
+            return a[0] * b[0];
+        }
 
         final double[] prodHigh = new double[len];
         double prodLowSum = 0;
