--- patch1-Closure-4-Hercules.buggys	2020-11-03 21:50:44.000000000 +0800
+++ patch1-Closure-4-Hercules.fixed	2020-11-03 21:52:28.000000000 +0800
@@ -187,7 +187,7 @@
     // makes more sense. Now, resolution via registry is first in order to
     // avoid triggering the warnings built into the resolution via properties.
     boolean resolved = resolveViaRegistry(t);
-    if (detectInheritanceCycle()) {
+    if (detectImplicitPrototypeCycle()) {
       handleTypeCycle(t);
     }
 
@@ -199,7 +199,7 @@
     }
 
     resolveViaProperties(t, enclosing);
-    if (detectInheritanceCycle()) {
+    if (detectImplicitPrototypeCycle()) {
       handleTypeCycle(t);
     }
 
