--- patch2-Chart-14-VFix.buggy	2020-11-01 19:12:47.000000000 +0800
+++ patch2-Chart-14-VFix.fixed	2020-11-01 19:13:06.000000000 +0800
@@ -2445,6 +2445,7 @@
             markers = (ArrayList) this.backgroundRangeMarkers.get(new Integer(
                     index));
         }
+        if(markers==null)   return false;
         boolean removed = markers.remove(marker);
         if (removed && notify) {
             fireChangeEvent();
