--- patch4-Chart-14-VFix.buggy	2020-11-01 19:19:24.000000000 +0800
+++ patch4-Chart-14-VFix.fixed	2020-11-01 19:19:37.000000000 +0800
@@ -2526,6 +2526,7 @@
             markers = (ArrayList) this.backgroundRangeMarkers.get(new Integer(
                     index));
         }
+        if(markers==null)   return false;
         boolean removed = markers.remove(marker);
         if (removed && notify) {
             fireChangeEvent();
