--- patch3-Chart-14-VFix.buggy	2020-11-01 19:16:57.000000000 +0800
+++ patch3-Chart-14-VFix.fixed	2020-11-01 19:17:07.000000000 +0800
@@ -2290,6 +2290,7 @@
             markers = (ArrayList) this.backgroundDomainMarkers.get(new Integer(
                     index));
         }
+        if(markers==null)   return false;
         boolean removed = markers.remove(marker);
         if (removed && notify) {
             fireChangeEvent();
