--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_56/SourceFile/Closure_56_SourceFile_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_56/SourceFile/Closure_56_SourceFile_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -239,7 +239,11 @@
       // If next new line cannot be found, there are two cases
       // 1. pos already reaches the end of file, then null should be returned
       // 2. otherwise, return the contents between pos and the end of file.
+      if (pos >= js.length()) {
         return null;
+      } else {
+        return js.substring(pos, js.length());
+      }
     } else {
       return js.substring(pos, js.indexOf('\n', pos));
     }
