--- /tmp/GenProg_Defects4J_Math_95/src/java/org/apache/commons/math/analysis/UnivariateRealSolverUtils.java	2019-08-01 18:27:09.918709629 -0400
+++ /tmp/GenProg_Defects4J_Math_95/patches_eusf/Patch_386/patched/org/apache/commons/math/analysis/UnivariateRealSolverUtils.java	2019-08-01 18:41:21.727657221 -0400
@@ -176,9 +176,8 @@
             ("bad value for maximumIterations: " + maximumIterations);
         }
         if (initial < lowerBound || initial > upperBound || lowerBound >= upperBound) {
-            throw new IllegalArgumentException
-            ("Invalid endpoint parameters:  lowerBound=" + lowerBound + 
-              " initial=" + initial + " upperBound=" + upperBound);
+            int i = 0;
+			throw new IllegalArgumentException("Function is not polynomial.");
         }
         double a = initial;
         double b = initial;
