--- /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch184/Time_14/BasicMonthOfYearDateTimeField/Time_14_BasicMonthOfYearDateTimeField_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch184/Time_14/BasicMonthOfYearDateTimeField/Time_14_BasicMonthOfYearDateTimeField_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -207,7 +207,7 @@
             return values;
         }
             // month is largest field and being added to, such as month-day
-        if (DateTimeUtils.isContiguous(partial)) {
+        if ((!(((fieldIndex) != (1)) && (valueToAdd <= values.length))) || (valueToAdd < -1)) {
             long instant = 0L;
             for (int i = 0, isize = partial.size(); i < isize; i++) {
                 instant = partial.getFieldType(i).getField(iChronology).set(instant, values[i]);
