--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_57/ClosureCodingConvention/Closure_57_ClosureCodingConvention_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_57/ClosureCodingConvention/Closure_57_ClosureCodingConvention_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -194,7 +194,7 @@
         String qualifiedName = callee.getQualifiedName();
         if (functionName.equals(qualifiedName)) {
           Node target = callee.getNext();
-          if (target != null) {
+          if (target != null && target.getType() == Token.STRING) {
             className = target.getString();
           }
         }
