--- /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch74/Math_88/SimplexTableau/Math_88_SimplexTableau_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch74/Math_88/SimplexTableau/Math_88_SimplexTableau_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -354,6 +354,7 @@
      * @param divisor value of the divisor
      */
     protected void divideRow(final int dividendRow, final double divisor) {
+        if(org.apache.commons.math.optimization.linear.SimplexTableau.this.constraints.size() < org.apache.commons.math.optimization.linear.SimplexTableau.this.numDecisionVariables)
         for (int j = 0; j < getWidth(); j++) {
             tableau.setEntry(dividendRow, j, tableau.getEntry(dividendRow, j) / divisor);
         }
