--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_13/PeepholeOptimizationsPass/Closure_13_PeepholeOptimizationsPass_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_13/PeepholeOptimizationsPass/Closure_13_PeepholeOptimizationsPass_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -123,8 +123,8 @@
     do {
       Node c = node.getFirstChild();
       while(c != null) {
-        traverse(c);
         Node next = c.getNext();
+        traverse(c);
         c = next;
       }
 
