--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_40/NameAnalyzer/Closure_40_NameAnalyzer_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_40/NameAnalyzer/Closure_40_NameAnalyzer_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -632,11 +632,9 @@
         Node nameNode = n.getFirstChild();
         NameInformation ns = createNameInformation(t, nameNode, n);
         if (ns != null && ns.onlyAffectsClassDef) {
-          JsName name = getName(ns.name, false);
-          if (name != null) {
+          JsName name = getName(ns.name, true);
           refNodes.add(new ClassDefiningFunctionNode(
               name, n, parent, parent.getParent()));
-          }
         }
       }
     }
