--- /Users/haoye.tian/Documents/University/data/PatchCollecting/jKali/Incorrect/Math/patch1-Math-49-jKali.buggy	2020-11-02 19:13:13.000000000 +0100
+++ /Users/haoye.tian/Documents/University/data/PatchCollecting/jKali/Incorrect/Math/patch1-Math-49-jKali.fixed	2020-11-02 19:13:13.000000000 +0100
@@ -658,16 +658,17 @@
         return (OpenMapRealVector) projection(new OpenMapRealVector(v));
     }
 
-    /** {@inheritDoc} */
+ 
     public void setEntry(int index, double value) {
         checkIndex(index);
-        if (!isDefaultValue(value)) {
-            entries.put(index, value);
-        } else if (entries.containsKey(index)) {
-            entries.remove(index);
+		if (true) {
+			this.entries.put(index, value);
+		}else 			if (this.entries.containsKey(index)) {
+				this.entries.remove(index);
         }
     }
 
+
     /** {@inheritDoc} */
     @Override
     public void setSubVector(int index, RealVector v) {
