--- a/src/java/org/apache/commons/cli2/commandline/WriteableCommandLineImpl.java
+++ b/src/java/org/apache/commons/cli2/commandline/WriteableCommandLineImpl.java
@@ -48,6 +48,8 @@ public class WriteableCommandLineImpl
     private final Map defaultSwitches = new HashMap();
     private final List normalised;
     private final Set prefixes;
+    private Option currentOption;
+    private String checkForOption;
 
     /**
      * Creates a new WriteableCommandLineImpl rooted on the specified Option, to
@@ -60,9 +62,16 @@ public class WriteableCommandLineImpl
                                     final List arguments) {
         this.prefixes = rootOption.getPrefixes();
         this.normalised = arguments;
+        setCurrentOption(rootOption);
     }
 
+    public Option getCurrentOption() {
+        return currentOption;
+    }
 
+    public void setCurrentOption(Option currentOption) {
+        this.currentOption = currentOption;
+    }
 
     public void addOption(Option option) {
         options.add(option);
@@ -231,18 +240,35 @@ public class WriteableCommandLineImpl
      */
     public boolean looksLikeOption(final String trigger)
     {
+        if (checkForOption != null)
+        {
             // this is a reentrant call
+            return !checkForOption.equals(trigger);
+        }
 
+        checkForOption = trigger;
+        try
+        {
             for (final Iterator i = prefixes.iterator(); i.hasNext();)
             {
                 final String prefix = (String) i.next();
 
                 if (trigger.startsWith(prefix))
                 {
+                    if (getCurrentOption().canProcess(this, trigger)
+                            || getCurrentOption().findOption(trigger) != null)
+                    {
                         return true;
+                    }
                 }
             }
+
             return false;
+        }
+        finally
+        {
+            checkForOption = null;
+        }
     }
 
     public String toString() {
