--- a/src/java/org/apache/commons/cli/CommandLine.java
+++ b/src/java/org/apache/commons/cli/CommandLine.java
@@ -16,11 +16,11 @@
 package org.apache.commons.cli;
 
 import java.util.Collection;
-import java.util.HashMap;
 import java.util.Iterator;
 import java.util.LinkedList;
 import java.util.List;
-import java.util.Map;
+import java.util.Set;
+import java.util.HashSet;
 
 /** 
  * <p>Represents list of arguments parsed against
@@ -43,12 +43,10 @@ public class CommandLine {
     private List args = new LinkedList();
 
     /** the processed options */
-    private Map options = new HashMap();
-    private Map names = new HashMap();
+    private Set options = new HashSet();
 
     /** Map of unique options for ease to get complete list of options */
 //    private Set allOptions = new HashSet();
-    private Map hashcodeMap = new HashMap();
 
     /**
      * Creates a command line.
@@ -66,7 +64,7 @@ public class CommandLine {
      */
     public boolean hasOption(String opt)
     {
-        return options.containsKey(opt);
+        return options.contains( resolveOption(opt));
     }
 
     /** 
@@ -90,12 +88,13 @@ public class CommandLine {
     {
         String res = getOptionValue(opt);
 
-        if (!options.containsKey(opt))
+        Option option = resolveOption(opt);
+        if (option == null)
         {
             return null;
         }
 
-        Object type = ((Option) options.get(opt)).getType();
+        Object type = option.getType();
 
         return (res == null)        ? null : TypeHandler.createValue(res, type);
     }
@@ -146,19 +145,13 @@ public class CommandLine {
      */
     public String[] getOptionValues(String opt)
     {
-        opt = Util.stripLeadingHyphens(opt);
-
-        String key = opt;
-        if (names.containsKey(opt))
+        Option key = resolveOption( opt );
 
+        if (options.contains(key))
         {
-            key = (String) names.get(opt);
+            return key.getValues();
         }
 
-        if (options.containsKey(key))
-        {
-            return ((Option) options.get(key)).getValues();
-        }
         return null;
         }
 
@@ -167,7 +160,24 @@ public class CommandLine {
      * @param opt short or long name of the option
      * @return Canonicalized option
      */
+    private Option resolveOption( String opt )
+    {
+        opt = Util.stripLeadingHyphens(opt);
+        for ( Iterator it = options.iterator(); it.hasNext(); )
+        {
+            Option option = (Option) it.next();
+            if (opt.equals(option.getOpt()))
+            {
+                return option;
+            }
+            if (opt.equals( option.getLongOpt()))
+            {
+                return option;
+        }
 
+        }
+        return null;
+    }
 
     /** 
      * Retrieves the array of values, if any, of an option.
@@ -275,17 +285,7 @@ public class CommandLine {
      */
     void addOption(Option opt)
     {
-        hashcodeMap.put(new Integer(opt.hashCode()), opt);
-        String key = opt.getKey();
-        if (key == null)
-        {
-            key = opt.getLongOpt();
-        }
-        else
-        {
-            names.put(opt.getLongOpt(), key);
-        }
-        options.put(key, opt);
+        options.add(opt);
     }
 
     /**
@@ -296,7 +296,7 @@ public class CommandLine {
      */
     public Iterator iterator()
     {
-        return hashcodeMap.values().iterator();
+        return options.iterator();
     }
 
     /**
@@ -306,7 +306,7 @@ public class CommandLine {
      */
     public Option[] getOptions()
     {
-        Collection processed = options.values();
+        Collection processed = options;
 
         // reinitialise array
         Option[] optionsArray = new Option[processed.size()];
@@ -314,4 +314,4 @@ public class CommandLine {
         // return the array
         return (Option[]) processed.toArray(optionsArray);
     }
-}
+}
\ No newline at end of file
