--- /Users/haoye.tian/Documents/University/data/PatchCollecting/Cardumen/Incorrect/Math/patch1-Math-85-Cardumen.buggy	2020-11-02 19:13:13.000000000 +0100
+++ /Users/haoye.tian/Documents/University/data/PatchCollecting/Cardumen/Incorrect/Math/patch1-Math-85-Cardumen.fixed	2020-11-02 19:13:13.000000000 +0100
@@ -175,13 +175,14 @@
     protected double getDomainLowerBound(double p) {
         double ret;
 
-        if (p < .5) {
-            ret = -Double.MAX_VALUE;
-        } else {
-            ret = getMean();
+ 		if (p < 0.5) {
+ 			ret = -(Double.MAX_VALUE);
+ 		}else {
+			ret = ((standardDeviation) - (standardDeviation)) / ((mean) * (Math.sqrt(2.0)));
         }
         
         return ret;
+
     }
 
     /**
