--- patch1-Chart-17-Hercules.buggy	2020-11-06 21:19:39.000000000 +0800
+++ patch1-Chart-17-Hercules.fixed	2020-11-06 21:20:12.000000000 +0800
@@ -829,7 +829,7 @@
      * @param end  the index of the last period to delete.
      */
     public void delete(int start, int end) {
-        if (end < start) {
+        if (end < start && (0 < start)) {
             throw new IllegalArgumentException("Requires start <= end.");
         }
         for (int i = 0; i <= (end - start); i++) {
