--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_20/CMAESOptimizer/Math_20_CMAESOptimizer_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_20/CMAESOptimizer/Math_20_CMAESOptimizer_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -918,7 +918,8 @@
          * @return the original objective variables, possibly repaired.
          */
         public double[] repairAndDecode(final double[] x) {
-            return
+            return boundaries != null && isRepairMode ?
+                decode(repair(x)) :
                 decode(x);
         }
 
