--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Chart_4/XYPlot/Chart_4_XYPlot_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Chart_4/XYPlot/Chart_4_XYPlot_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -4490,6 +4490,7 @@
                     }
                 }
                 
+                if (r != null) {
                     Collection c = r.getAnnotations();
                     Iterator i = c.iterator();
                     while (i.hasNext()) {
@@ -4498,6 +4499,7 @@
                             includedAnnotations.add(a);
                         }
                     }
+                }
             }
         }
 
