--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_35/ElitisticListPopulation/Math_35_ElitisticListPopulation_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_35/ElitisticListPopulation/Math_35_ElitisticListPopulation_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -48,7 +48,7 @@
                                    final int populationLimit,
                                    final double elitismRate) {
         super(chromosomes, populationLimit);
-        this.elitismRate = elitismRate;
+        setElitismRate(elitismRate);
     }
 
     /**
@@ -62,7 +62,7 @@
      */
     public ElitisticListPopulation(final int populationLimit, final double elitismRate) {
         super(populationLimit);
-        this.elitismRate = elitismRate;
+        setElitismRate(elitismRate);
     }
 
     /**
