--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Time_24/DateTimeParserBucket/Time_24_DateTimeParserBucket_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Time_24/DateTimeParserBucket/Time_24_DateTimeParserBucket_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -352,6 +352,11 @@
             for (int i = 0; i < count; i++) {
                 millis = savedFields[i].set(millis, resetFields);
             }
+            if (resetFields) {
+                for (int i = 0; i < count; i++) {
+                    millis = savedFields[i].set(millis, i == (count - 1));
+                }
+            }
         } catch (IllegalFieldValueException e) {
             if (text != null) {
                 e.prependMessage("Cannot parse \"" + text + '"');
