--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_14/ControlFlowAnalysis/Closure_14_ControlFlowAnalysis_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_14/ControlFlowAnalysis/Closure_14_ControlFlowAnalysis_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -764,7 +764,7 @@
         } else if (parent.getLastChild() == node){
           if (cfa != null) {
             for (Node finallyNode : cfa.finallyMap.get(parent)) {
-              cfa.createEdge(fromNode, Branch.UNCOND, finallyNode);
+              cfa.createEdge(fromNode, Branch.ON_EX, finallyNode);
             }
           }
           return computeFollowNode(fromNode, parent, cfa);
