--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_114/NameAnalyzer/Closure_114_NameAnalyzer_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_114/NameAnalyzer/Closure_114_NameAnalyzer_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -575,7 +575,7 @@
           } else {
             recordDepScope(nameNode, ns);
           }
-        } else {
+        } else if (!(parent.isCall() && parent.getFirstChild() == n)) {
           // The rhs of the assignment is the caller, so it's used by the
           // context. Don't associate it w/ the lhs.
           // FYI: this fixes only the specific case where the assignment is the
