--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_65/CodeGenerator/Closure_65_CodeGenerator_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_65/CodeGenerator/Closure_65_CodeGenerator_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -1012,7 +1012,7 @@
     for (int i = 0; i < s.length(); i++) {
       char c = s.charAt(i);
       switch (c) {
-        case '\0': sb.append("\\0"); break;
+        case '\0': sb.append("\\000"); break;
         case '\n': sb.append("\\n"); break;
         case '\r': sb.append("\\r"); break;
         case '\t': sb.append("\\t"); break;
