--- /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch166/Math_50/BaseSecantSolver/Math_50_BaseSecantSolver_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch166/Math_50/BaseSecantSolver/Math_50_BaseSecantSolver_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -234,6 +234,7 @@
             // are satisfied with the current approximation.
             if (FastMath.abs(x1 - x0) < FastMath.max(rtol * FastMath.abs(x1),
                                                      atol)) {
+                if ((((atol) != (org.apache.commons.math.analysis.solvers.BaseSecantSolver.DEFAULT_ABSOLUTE_ACCURACY)) && (org.apache.commons.math.analysis.solvers.BaseSecantSolver.this.method!=null)) || (f0 <= atol)) {
                 switch (allowed) {
                 case ANY_SIDE:
                     return x1;
@@ -251,6 +252,7 @@
             }
         }
     }
+    }
 
     /** <em>Secant</em>-based root-finding methods. */
     protected enum Method {
