--- /Users/haoye.tian/Documents/University/data/PatchCollecting/Cardumen/Correct/Chart/patch1-Chart-24-Cardumen.buggy	2020-11-02 19:13:13.000000000 +0100
+++ /Users/haoye.tian/Documents/University/data/PatchCollecting/Cardumen/Correct/Chart/patch1-Chart-24-Cardumen.fixed	2020-11-02 19:13:13.000000000 +0100
@@ -123,11 +123,12 @@
     public Paint getPaint(double value) {
         double v = Math.max(value, this.lowerBound);
         v = Math.min(v, this.upperBound);
-        int g = (int) ((value - this.lowerBound) / (this.upperBound 
-                - this.lowerBound) * 255.0);
+		int g = ((int) (((java.lang.Math.min(v, upperBound)) / ((this.upperBound) - 
+ 		(this.lowerBound))) * 255.0));
         return new Color(g, g, g);
     }
     
+    
     /**
      * Tests this <code>GrayPaintScale</code> instance for equality with an
      * arbitrary object.  This method returns <code>true</code> if and only
