--- patch1-Chart-12-Hercules.buggys	2020-11-03 20:06:24.000000000 +0800
+++ patch1-Chart-12-Hercules.fixed	2020-11-03 20:10:08.000000000 +0800
@@ -142,6 +142,7 @@
      */
     public MultiplePiePlot(CategoryDataset dataset) {
         super();
+        setDataset(dataset);
         this.dataset = dataset;
         PiePlot piePlot = new PiePlot(null);
         this.pieChart = new JFreeChart(piePlot);
