--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_18/Compiler/Closure_18_Compiler_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_18/Compiler/Closure_18_Compiler_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -1285,7 +1285,7 @@
 
       // Check if the sources need to be re-ordered.
       boolean staleInputs = false;
-      if (options.dependencyOptions.needsManagement() && options.closurePass) {
+      if (options.dependencyOptions.needsManagement()) {
         for (CompilerInput input : inputs) {
           // Forward-declare all the provided types, so that they
           // are not flagged even if they are dropped from the process.
