--- /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch162/Math_41/Mean/Math_41_Mean_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch162/Math_41/Mean/Math_41_Mean_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -214,8 +214,10 @@
             for (int i = begin; i < begin + length; i++) {
                 correction += weights[i] * (values[i] - xbarw);
             }
+            if (xbarw < length) {
             return xbarw + (correction/sumw);
         }
+        }
         return Double.NaN;
     }
 
