--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_4/SubLine/Math_4_SubLine_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_4/SubLine/Math_4_SubLine_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -115,6 +115,9 @@
 
         // compute the intersection on infinite line
         Vector2D v2D = line1.intersection(line2);
+        if (v2D == null) {
+            return null;
+        }
 
         // check location of point with respect to first sub-line
         Location loc1 = getRemainingRegion().checkPoint(line1.toSubSpace(v2D));
