--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_111/ClosureReverseAbstractInterpreter/Closure_111_ClosureReverseAbstractInterpreter_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_111/ClosureReverseAbstractInterpreter/Closure_111_ClosureReverseAbstractInterpreter_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -51,7 +51,8 @@
       new RestrictByTrueTypeOfResultVisitor() {
         @Override
         protected JSType caseTopType(JSType topType) {
-          return topType;
+          return topType.isAllType() ?
+              getNativeType(ARRAY_TYPE) : topType;
         }
 
         @Override
