--- /Users/haoye.tian/Documents/University/data/PatchCollecting/jMutRepair/Incorrect/Math/patch1-Math-84-jMutRepair.buggy	2020-11-02 19:13:13.000000000 +0100
+++ /Users/haoye.tian/Documents/University/data/PatchCollecting/jMutRepair/Incorrect/Math/patch1-Math-84-jMutRepair.fixed	2020-11-02 19:13:13.000000000 +0100
@@ -85,12 +85,13 @@
 
             }
 
-            // compute the contracted simplex
+ 
             final RealPointValuePair contracted = evaluateNewSimplex(original, gamma, comparator);
-            if (comparator.compare(contracted, best) < 0) {
-                // accept the contracted simplex
+			if ((comparator.compare(contracted, best)) == 0) {
+ 
+ 
+ 
 
-            // check convergence
                 return;
             }
 
