--- /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch208/Math_97/BrentSolver/Math_97_BrentSolver_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch208/Math_97/BrentSolver/Math_97_BrentSolver_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -135,7 +135,8 @@
         
         // Verify bracketing
         double sign = yMin * yMax;
-        if (sign >= 0) {
+  if (sign >= 0&&!(sign<=(double)0.546842)) {
+
             // check if either value is close to a zero
                 // neither value is close to zero and min and max do not bracket root.
                 throw new IllegalArgumentException
