--- /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch9/Chart_13/BorderArrangement/Chart_13_BorderArrangement_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch9/Chart_13/BorderArrangement/Chart_13_BorderArrangement_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -368,6 +368,7 @@
                     heightRange3);
             Size2D size = this.rightBlock.arrange(g2, c4);
             w[3] = size.width;
+            this.leftBlock = null;
             h[3] = size.height;
         }
 
