--- patch1-Lang-47-Hercules.buggys	2020-11-04 00:03:20.000000000 +0800
+++ patch1-Lang-47-Hercules.fixed	2020-11-04 00:06:32.000000000 +0800
@@ -1183,7 +1183,11 @@
         if (width > 0) {
             ensureCapacity(size + width);
             String str = (obj == null ? getNullText() : obj.toString());
-            int strLen = str.length();
+            int strLen = 0;
+            if ((str != null))
+                strLen = str.length();
+            else
+                str = ""
             if (strLen >= width) {
                 str.getChars(strLen - width, strLen, buffer, size);
             } else {
@@ -1227,7 +1227,11 @@
         if (width > 0) {
             ensureCapacity(size + width);
             String str = (obj == null ? getNullText() : obj.toString());
-            int strLen = str.length();
+            int strLen = 0;
+            if ((str != null))
+                strLen = str.length();
+            else
+                str = "";
             if (strLen >= width) {
                 str.getChars(0, width, buffer, size);
             } else {