--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_82/JSType/Closure_82_JSType_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_82/JSType/Closure_82_JSType_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -160,7 +160,9 @@
   }
 
   public final boolean isEmptyType() {
-    return isNoType() || isNoObjectType() || isNoResolvedType();
+    return isNoType() || isNoObjectType() || isNoResolvedType() ||
+        (registry.getNativeFunctionType(
+             JSTypeNative.LEAST_FUNCTION_TYPE) == this);
   }
 
   public boolean isNumberObjectType() {
