--- /tmp/GenProg_Defects4J_Closure_75/src/com/google/javascript/jscomp/PeepholeFoldConstants.java	2019-08-11 16:19:02.642625975 -0400
+++ /tmp/GenProg_Defects4J_Closure_75/patches_mgrb/Patch_47/patched/com/google/javascript/jscomp/PeepholeFoldConstants.java	2019-08-11 16:42:37.799316025 -0400
@@ -260,9 +260,6 @@
     } else {
       replacement = Node.newNumber(value);
     }
-
-    n.getParent().replaceChild(n, replacement);
-    reportCodeChange();
   }
 
   /**
