--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_131/TokenStream/Closure_131_TokenStream_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_131/TokenStream/Closure_131_TokenStream_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -191,12 +191,13 @@
       int length = s.length();
 
       if (length == 0 ||
+          Character.isIdentifierIgnorable(s.charAt(0)) ||
           !Character.isJavaIdentifierStart(s.charAt(0))) {
         return false;
       }
 
       for (int i = 1; i < length; i++) {
-        if (
+        if (Character.isIdentifierIgnorable(s.charAt(i)) ||
             !Character.isJavaIdentifierPart(s.charAt(i))) {
           return false;
         }
