--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_52/CodeGenerator/Closure_52_CodeGenerator_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_52/CodeGenerator/Closure_52_CodeGenerator_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -742,7 +742,7 @@
         return false;
       }
     }
-    return len > 0;
+    return len > 0 && s.charAt(0) != '0';
   }
 
   static double getSimpleNumber(String s) {
