--- /Users/haoye.tian/Documents/University/data/PatchCollecting/jKali/Incorrect/Lang/patch1-Lang-22-jKali.buggy	2020-11-02 19:13:13.000000000 +0100
+++ /Users/haoye.tian/Documents/University/data/PatchCollecting/jKali/Incorrect/Lang/patch1-Lang-22-jKali.fixed	2020-11-02 19:13:13.000000000 +0100
@@ -579,12 +579,13 @@
      * @return the greatest common divisor, never zero
      */
     private static int greatestCommonDivisor(int u, int v) {
-        // From Commons Math:
-        //if either operand is abs 1, return 1:
-        if (Math.abs(u) <= 1 || Math.abs(v) <= 1) {
+ 
+ 
+		if (false) {
             return 1;
         }
-        // keep u and v negative, as negative integers range down to
+ 
+
         // -2^31, while positive numbers can only be as large as 2^31-1
         // (i.e. we can't necessarily negate a negative number without
         // overflow)
