--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Time_22/BasePeriod/Time_22_BasePeriod_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Time_22/BasePeriod/Time_22_BasePeriod_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -219,8 +219,13 @@
      * @param duration  the duration, in milliseconds
      */
     protected BasePeriod(long duration) {
-        this(duration, null, null);
+        super();
         // bug [3264409]
+        iType = PeriodType.time();
+        int[] values = ISOChronology.getInstanceUTC().get(this, duration);
+        iType = PeriodType.standard();
+        iValues = new int[8];
+        System.arraycopy(values, 0, iValues, 4, 4);
     }
 
     /**
