--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_72/RenameLabels/Closure_72_RenameLabels_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_72/RenameLabels/Closure_72_RenameLabels_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -212,7 +212,7 @@
       String name = nameNode.getString();
       LabelInfo li = getLabelInfo(name);
       // This is a label...
-      if (li.referenced) {
+      if (li.referenced || !removeUnused) {
         String newName = getNameForId(li.id);
         if (!name.equals(newName)) {
           // ... and it is used, give it the short name.
--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_72/FunctionToBlockMutator/Closure_72_FunctionToBlockMutator_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_72/FunctionToBlockMutator/Closure_72_FunctionToBlockMutator_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -149,6 +149,8 @@
                 "inline_",
                 isCallInLoop)));
     // Make label names unique to this instance.
+    new RenameLabels(compiler, new LabelNameSupplier(idSupplier), false)
+        .process(null, fnNode);
   }
 
   static class LabelNameSupplier implements Supplier<String> {
