--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Lang_8/FastDatePrinter/Lang_8_FastDatePrinter_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Lang_8/FastDatePrinter/Lang_8_FastDatePrinter_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -1095,7 +1095,6 @@
     private static class TimeZoneNameRule implements Rule {
         private final Locale mLocale;
         private final int mStyle;
-        private final TimeZone zone;
         private final String mStandard;
         private final String mDaylight;
 
@@ -1109,7 +1108,6 @@
         TimeZoneNameRule(TimeZone timeZone, Locale locale, int style) {
             mLocale = locale;
             mStyle = style;
-            zone = timeZone;
             
             mStandard = getTimeZoneDisplay(timeZone, false, style, locale);
             mDaylight = getTimeZoneDisplay(timeZone, true, style, locale);
@@ -1131,6 +1129,7 @@
          */
         @Override
         public void appendTo(StringBuffer buffer, Calendar calendar) {
+            TimeZone zone = calendar.getTimeZone();
             if (zone.useDaylightTime()
                     && calendar.get(Calendar.DST_OFFSET) != 0) {
                 buffer.append(getTimeZoneDisplay(zone, true, mStyle, mLocale));
