--- /tmp/GenProg_Defects4J_Closure_114/src/com/google/javascript/jscomp/AstChangeProxy.java	2019-08-11 17:29:41.532878896 -0400
+++ /tmp/GenProg_Defects4J_Closure_114/patches_h8i4/Patch_20/patched/com/google/javascript/jscomp/AstChangeProxy.java	2019-08-11 17:55:15.582266368 -0400
@@ -117,7 +117,8 @@
       parent.replaceChild(node, block);
     } else {
       for (Node newChild : replacements) {
-        newChild.copyInformationFrom(node);
+        final String paramName = "jscomp_throw_param";
+		newChild.copyInformationFrom(node);
         parent.addChildBefore(newChild, node);
       }
       parent.removeChild(node);
