--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Lang_11/RandomStringUtils/Lang_11_RandomStringUtils_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Lang_11/RandomStringUtils/Lang_11_RandomStringUtils_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -242,6 +242,10 @@
                     start = ' ';                
                 }
             }
+        } else {
+            if (end <= start) {
+                throw new IllegalArgumentException("Parameter end (" + end + ") must be greater than start (" + start + ")");
+            }
         }
 
         char[] buffer = new char[count];
