--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Lang_48/EqualsBuilder/Lang_48_EqualsBuilder_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Lang_48/EqualsBuilder/Lang_48_EqualsBuilder_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -377,8 +377,12 @@
         }
         Class lhsClass = lhs.getClass();
         if (!lhsClass.isArray()) {
+            if (lhs instanceof java.math.BigDecimal) {
+                isEquals = (((java.math.BigDecimal)lhs).compareTo(rhs) == 0);
+            } else {
                 // The simple case, not an array, just test the element
                 isEquals = lhs.equals(rhs);
+            }
         } else if (lhs.getClass() != rhs.getClass()) {
             // Here when we compare different dimensions, for example: a boolean[][] to a boolean[] 
             this.setEquals(false);
