--- patch2-Lang-47-VFix.buggy	2020-11-01 20:40:56.000000000 +0800
+++ patch2-Lang-47-VFix.fixed	2020-11-01 20:41:07.000000000 +0800
@@ -1227,6 +1227,7 @@
         if (width > 0) {
             ensureCapacity(size + width);
             String str = (obj == null ? getNullText() : obj.toString());
+            if(str==null)   str="";
             int strLen = str.length();
             if (strLen >= width) {
                 str.getChars(0, width, buffer, size);
