--- patch1-Lang-47-VFix.buggy	2020-11-01 20:39:19.000000000 +0800
+++ patch1-Lang-47-VFix.fixed	2020-11-01 20:39:30.000000000 +0800
@@ -1183,6 +1183,7 @@
         if (width > 0) {
             ensureCapacity(size + width);
             String str = (obj == null ? getNullText() : obj.toString());
+            if(str==null)   str="";
             int strLen = str.length();
             if (strLen >= width) {
                 str.getChars(strLen - width, strLen, buffer, size);
