--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_60/NormalDistributionImpl/Math_60_NormalDistributionImpl_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_60/NormalDistributionImpl/Math_60_NormalDistributionImpl_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -123,18 +123,11 @@
      */
     public double cumulativeProbability(double x) throws MathException {
         final double dev = x - mean;
-        try {
+        if (FastMath.abs(dev) > 40 * standardDeviation) { 
+            return dev < 0 ? 0.0d : 1.0d;
+        }
         return 0.5 * (1.0 + Erf.erf((dev) /
                     (standardDeviation * FastMath.sqrt(2.0))));
-        } catch (MaxIterationsExceededException ex) {
-            if (x < (mean - 20 * standardDeviation)) { // JDK 1.5 blows at 38
-                return 0;
-            } else if (x > (mean + 20 * standardDeviation)) {
-                return 1;
-            } else {
-                throw ex;
-            }
-        }
     }
 
     /**
