--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_123/CodeGenerator/Closure_123_CodeGenerator_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_123/CodeGenerator/Closure_123_CodeGenerator_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -282,7 +282,7 @@
       case Token.HOOK: {
         Preconditions.checkState(childCount == 3);
         int p = NodeUtil.precedence(type);
-        Context rhsContext = Context.OTHER;
+        Context rhsContext = getContextForNoInOperator(context);
         addExpr(first, p + 1, context);
         cc.addOp("?", true);
         addExpr(first.getNext(), 1, rhsContext);
