--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Lang_62/Entities/Lang_62_Entities_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Lang_62/Entities/Lang_62_Entities_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -847,6 +847,9 @@
                             } else {
                                 entityValue = Integer.parseInt(entityName.substring(1));
                             }
+                            if (entityValue > 0xFFFF) {
+                                entityValue = -1;
+                            }
                         } catch (NumberFormatException ex) {
                             entityValue = -1;
                         }
@@ -917,12 +920,17 @@
                                     case 'X' :
                                     case 'x' : {
                                         entityValue = Integer.parseInt(entityContent.substring(2), 16);
+                                        break;
                                     }
                                     default : {
                                         entityValue = Integer.parseInt(entityContent.substring(1), 10);
                                     }
                                 }
+                                if (entityValue > 0xFFFF) {
+                                    entityValue = -1;
+                                }
                             } catch (NumberFormatException e) {
+                                entityValue = -1;
                             }
                         }
                     } else { //escaped value content is an entity name
