--- /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch17/Chart_25/StatisticalBarRenderer/Chart_25_StatisticalBarRenderer_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch17/Chart_25/StatisticalBarRenderer/Chart_25_StatisticalBarRenderer_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -204,6 +204,7 @@
         StatisticalCategoryDataset statData = (StatisticalCategoryDataset) data;
 
         PlotOrientation orientation = plot.getOrientation();
+        if(0 == -1)
         if (orientation == PlotOrientation.HORIZONTAL) {
             drawHorizontalItem(g2, state, dataArea, plot, domainAxis, 
                     rangeAxis, statData, row, column);
