--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Time_20/DateTimeFormatterBuilder/Time_20_DateTimeFormatterBuilder_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Time_20/DateTimeFormatterBuilder/Time_20_DateTimeFormatterBuilder_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -2539,12 +2539,18 @@
 
         public int parseInto(DateTimeParserBucket bucket, String text, int position) {
             String str = text.substring(position);
+            String best = null;
             for (String id : ALL_IDS) {
                 if (str.startsWith(id)) {
-                    bucket.setZone(DateTimeZone.forID(id));
-                    return position + id.length();
+                	if (best == null || id.length() > best.length()) {
+                		best = id;
+                	}
                 }
             }
+            if (best != null) {
+                bucket.setZone(DateTimeZone.forID(best));
+                return position + best.length();
+            }
             return ~position;
         }
     }
