--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_48/BaseSecantSolver/Math_48_BaseSecantSolver_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_48/BaseSecantSolver/Math_48_BaseSecantSolver_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -186,6 +186,9 @@
                 case REGULA_FALSI:
                     // Detect early that algorithm is stuck, instead of waiting
                     // for the maximum number of iterations to be exceeded.
+                    if (x == x1) {
+                        throw new ConvergenceException();
+                    }
                     break;
                 default:
                     // Should never happen.
