--- patch1-Closure-109-Hercules.buggys	2020-11-03 23:37:25.000000000 +0800
+++ patch1-Closure-109-Hercules.fixed	2020-11-03 23:38:02.000000000 +0800
@@ -1905,7 +1905,7 @@
    * For expressions on the right hand side of a this: or new:
    */
   private Node parseContextTypeExpression(JsDocToken token) {
-          return parseTypeName(token);
+          return parseTypeExpression(token);
   }
 
   /**
