--- patch2-Lang-20-VFix.buggy	2020-11-01 20:32:42.000000000 +0800
+++ patch2-Lang-20-VFix.fixed	2020-11-01 20:33:05.000000000 +0800
@@ -3378,8 +3378,10 @@
         int noOfItems = (endIndex - startIndex);
         if (noOfItems <= 0) {
             return EMPTY;
-        }  
-        StringBuilder buf = new StringBuilder((array[startIndex] == null ? 16 : array[startIndex].toString().length()) + separator.length());
+        }
+        String str=array[startIndex].toString();
+        if(str==null) str="";    
+        StringBuilder buf = new StringBuilder((array[startIndex] == null ? 16 : str.length()) + separator.length());
         StringBuilder buf = new StringBuilder((array[startIndex] == null ? 16 : array[startIndex].toString().length()) + separator.length());
 
         for (int i = startIndex; i < endIndex; i++) {
