--- patch1-Lang-20-VFix.buggy	2020-11-01 20:28:44.000000000 +0800
+++ patch1-Lang-20-VFix.fixed	2020-11-01 20:29:07.000000000 +0800
@@ -3294,7 +3294,9 @@
         if (noOfItems <= 0) {
             return EMPTY;
         }
-        StringBuilder buf = new StringBuilder((array[startIndex] == null ? 16 : array[startIndex].toString().length()) + 1);        
+        String str=array[startIndex].toString();
+        if(str==null) str="";    
+        StringBuilder buf = new StringBuilder((array[startIndex] == null ? 16 : str.length()) + 1);        
         StringBuilder buf = new StringBuilder((array[startIndex] == null ? 16 : array[startIndex].toString().length()) + 1);
 
         for (int i = startIndex; i < endIndex; i++) {
