--- patch1-Math-4-Hercules.buggy	2020-11-06 19:04:38.000000000 +0800
+++ patch1-Math-4-Hercules.fixed	2020-11-06 19:06:56.000000000 +0800
@@ -113,6 +113,9 @@
         Vector3D v1D = line.intersection(subLine.line);
 
         // check location of point with respect to first sub-line
+        if ((v1D == null) || (remainingRegion == null) || (line == null))
+            return null;
+            
         Location loc1 = remainingRegion.checkPoint(line.toSubSpace(v1D));
 
         // check location of point with respect to second sub-line
