--- patch1-Chart-26-VFix.buggy	2020-11-01 20:07:20.000000000 +0800
+++ patch1-Chart-26-VFix.fixed	2020-11-01 20:07:33.000000000 +0800
@@ -1189,12 +1189,14 @@
         }
         if (plotState != null && hotspot != null) {
             ChartRenderingInfo owner = plotState.getOwner();
+        if(owner!=null){
                 EntityCollection entities = owner.getEntityCollection();
                 if (entities != null) {
                     entities.add(new AxisLabelEntity(this, hotspot, 
                             this.labelToolTip, this.labelURL));
                 }
         }
+        }
         return state;
 
     }
