--- patch1-Time-14-Hercules.buggy	2020-11-06 20:57:15.000000000 +0800
+++ patch1-Time-14-Hercules.fixed	2020-11-06 20:57:51.000000000 +0800
@@ -516,7 +516,7 @@
             return this;
         }
         int[] newValues = getValues();
-        newValues = getField(index).add(this, index, newValues, amount);
+        newValues = getField(index).addWrapPartial(this, index, newValues, amount);"
         return new MonthDay(this, newValues);
     }
 
