--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_43/TypedScopeCreator/Closure_43_TypedScopeCreator_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_43/TypedScopeCreator/Closure_43_TypedScopeCreator_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -417,6 +417,7 @@
      * For more information, see
      * http://code.google.com/p/closure-compiler/issues/detail?id=314
      */
+    private List<Node> lentObjectLiterals = null;
 
     /**
      * Type-less stubs.
@@ -544,6 +545,13 @@
       }
 
       // Analyze any @lends object literals in this statement.
+      if (n.getParent() != null && NodeUtil.isStatement(n) &&
+          lentObjectLiterals != null) {
+        for (Node objLit : lentObjectLiterals) {
+          defineObjectLiteral(objLit);
+        }
+        lentObjectLiterals.clear();
+      }
     }
 
     private void attachLiteralTypes(NodeTraversal t, Node n) {
@@ -577,7 +585,16 @@
           break;
 
         case Token.OBJECTLIT:
+          JSDocInfo info = n.getJSDocInfo();
+          if (info != null &&
+              info.getLendsName() != null) {
+            if (lentObjectLiterals == null) {
+              lentObjectLiterals = Lists.newArrayList();
+            }
+            lentObjectLiterals.add(n);
+          } else {
             defineObjectLiteral(n);
+          }
           break;
 
           // NOTE(nicksantos): If we ever support Array tuples,
