--- NameAnalyzer.java	2020-10-10 17:31:41.000000000 +0200
+++ NameAnalyzer copy.java	2020-11-07 14:48:27.000000000 +0100
@@ -632,7 +632,7 @@
         Node nameNode = n.getFirstChild();
         NameInformation ns = createNameInformation(t, nameNode, n);
         if (ns != null && ns.onlyAffectsClassDef) {
-          JsName name = getName(ns.name, false);
+          JsName name = getName(ns.name, true);
           if (name != null) {
           refNodes.add(new ClassDefiningFunctionNode(
               name, n, parent, parent.getParent()));
