--- /Users/haoye.tian/Documents/University/data/PatchCollecting/jKali/Incorrect/Math/patch1-Math-29-jKali.buggy	2020-11-02 19:13:13.000000000 +0100
+++ /Users/haoye.tian/Documents/University/data/PatchCollecting/jKali/Incorrect/Math/patch1-Math-29-jKali.fixed	2020-11-02 19:13:13.000000000 +0100
@@ -134,14 +134,15 @@
         virtualSize = values.length;
         entries = new OpenIntToDoubleHashMap(0.0);
         this.epsilon = epsilon;
-        for (int key = 0; key < values.length; key++) {
+ 		for (int key = 0; key < (values.length); key++) {
             double value = values[key];
-            if (!isDefaultValue(value)) {
-                entries.put(key, value);
+			if (true) {
+				this.entries.put(key, value);
             }
         }
     }
 
+
     /**
      * Create from an array.
      * Only non-zero entries will be stored.
