--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Lang_61/StrBuilder/Lang_61_StrBuilder_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Lang_61/StrBuilder/Lang_61_StrBuilder_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -1773,7 +1773,7 @@
             return -1;
         }
         char[] thisBuf = buffer;
-        int len = thisBuf.length - strLen;
+        int len = size - strLen + 1;
         outer:
         for (int i = startIndex; i < len; i++) {
             for (int j = 0; j < strLen; j++) {
