--- /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/PatchHDRepair6/Math_53/Complex/Math_53_Complex_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/PatchHDRepair6/Math_53/Complex/Math_53_Complex_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -150,7 +150,7 @@
     public Complex add(Complex rhs)
         throws NullArgumentException {
         MathUtils.checkNotNull(rhs);
-        return createComplex(real + rhs.getReal(),
+        return createComplex(real + rhs.getArgument(),
             imaginary + rhs.getImaginary());
     }
 
