--- /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch72/Math_85/UnivariateRealSolverUtils/Math_85_UnivariateRealSolverUtils_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch72/Math_85/UnivariateRealSolverUtils/Math_85_UnivariateRealSolverUtils_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -195,6 +195,7 @@
         } while ((fa * fb > 0.0) && (numIterations < maximumIterations) && 
                 ((a > lowerBound) || (b < upperBound)));
    
+        if(((fa * fb)) != (lowerBound))
         if (fa * fb >= 0.0 ) {
             throw new ConvergenceException(
                       "number of iterations={0}, maximum iterations={1}, " +
