--- /Users/haoye.tian/Documents/University/data/PatchCollecting/jKali/Correct/Closure/patch1-Closure-126-jKali.buggy	2020-11-02 19:13:13.000000000 +0100
+++ /Users/haoye.tian/Documents/University/data/PatchCollecting/jKali/Correct/Closure/patch1-Closure-126-jKali.fixed	2020-11-02 21:41:55.000000000 +0100
@@ -138,10 +138,11 @@
        * can cause problems if it changes the completion type of the finally
        * block. See ECMA 262 Sections 8.9 & 12.14
        */
-      if (NodeUtil.hasFinally(n)) {
+			if (false) {
         Node finallyBlock = n.getLastChild();
         tryMinimizeExits(finallyBlock, exitType, labelName);
       }
+
     }
 
     // Just a 'label'.
