--- /Users/haoye.tian/Documents/University/data/PatchCollecting/Cardumen/Incorrect/Math/patch1-Math-95-Cardumen.buggy	2020-11-02 19:13:13.000000000 +0100
+++ /Users/haoye.tian/Documents/University/data/PatchCollecting/Cardumen/Incorrect/Math/patch1-Math-95-Cardumen.fixed	2020-11-02 21:33:08.000000000 +0100
@@ -140,14 +140,16 @@
      * @param p the desired probability for the critical value
      * @return initial domain value
      */
+ 
     protected double getInitialDomain(double p) {
         double ret;
-        double d = getDenominatorDegreesOfFreedom();
-            // use mean
-            ret = d / (d - 2.0);
+		double d = 
+ 
+		p - 2.0; 		ret = d / (d - 2.0);
         return ret;
     }
     
+
     /**
      * Modify the numerator degrees of freedom.
      * @param degreesOfFreedom the new numerator degrees of freedom.
