--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_31/Compiler/Closure_31_Compiler_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_31/Compiler/Closure_31_Compiler_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -1282,7 +1282,6 @@
 
       // Check if the sources need to be re-ordered.
       if (options.dependencyOptions.needsManagement() &&
-          !options.skipAllPasses &&
           options.closurePass) {
         for (CompilerInput input : inputs) {
           // Forward-declare all the provided types, so that they
