--- patch1-Lang-38-Hercules.buggys	2020-11-03 23:56:09.000000000 +0800
+++ patch1-Lang-38-Hercules.fixed	2020-11-03 23:57:30.000000000 +0800
@@ -869,6 +869,7 @@
      */
     public StringBuffer format(Calendar calendar, StringBuffer buf) {
         if (mTimeZoneForced) {
+            calendar.getTime();
             calendar.getTimeInMillis(); /// LANG-538
             calendar = (Calendar) calendar.clone();
             calendar.setTimeZone(mTimeZone);
