--- /Users/haoye.tian/Downloads/ODS/data/PS/correct/Patch26/Lang_58/NumberUtils/Lang_58_NumberUtils_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/PS/correct/Patch26/Lang_58/NumberUtils/Lang_58_NumberUtils_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -461,6 +461,7 @@
                         return createBigInteger(numeric);
 
                     }
+                    if(-1 < expPos)
                     throw new NumberFormatException(str + " is not a valid number.");
                 case 'f' :
                 case 'F' :
