--- /Users/haoye.tian/Documents/University/data/PatchCollecting/jMutRepair/Correct/Math/patch1-Math-85-jMutRepair.buggy	2020-11-02 19:13:13.000000000 +0100
+++ /Users/haoye.tian/Documents/University/data/PatchCollecting/jMutRepair/Correct/Math/patch1-Math-85-jMutRepair.fixed	2020-11-02 19:13:13.000000000 +0100
@@ -192,13 +192,14 @@
             
             fb = function.value(b);
             numIterations++ ;
-        } while ((fa * fb > 0.0) && (numIterations < maximumIterations) && 
-                ((a > lowerBound) || (b < upperBound)));
+ 		} while ((((fa * fb) > 0.0) && (numIterations < maximumIterations)) && 
+ 		((a > lowerBound) || (b < upperBound)) );
+ 
+		if ((fa * fb) > 0.0) {
+ 			throw new math.ConvergenceException(
+ 			("number of iterations={0}, maximum iterations={1}, " + 
+ 			("initial={2}, lower bound={3}, upper bound={4}, final a value={5}, " + 
    
-        if (fa * fb >= 0.0 ) {
-            throw new ConvergenceException(
-                      "number of iterations={0}, maximum iterations={1}, " +
-                      "initial={2}, lower bound={3}, upper bound={4}, final a value={5}, " +
                       "final b value={6}, f(a)={7}, f(b)={8}",
                       numIterations, maximumIterations, initial,
                       lowerBound, upperBound, a, b, fa, fb);
