--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Time_15/FieldUtils/Time_15_FieldUtils_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Time_15/FieldUtils/Time_15_FieldUtils_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -135,6 +135,9 @@
     public static long safeMultiply(long val1, int val2) {
         switch (val2) {
             case -1:
+                if (val1 == Long.MIN_VALUE) {
+                    throw new ArithmeticException("Multiplication overflows a long: " + val1 + " * " + val2);
+                }
                 return -val1;
             case 0:
                 return 0L;
