--- /Users/haoye.tian/Documents/University/data/PatchCollecting/jKali/Incorrect/Closure/patch1-Closure-62-jKali.buggy	2020-11-02 19:13:13.000000000 +0100
+++ /Users/haoye.tian/Documents/University/data/PatchCollecting/jKali/Incorrect/Closure/patch1-Closure-62-jKali.fixed	2020-11-02 19:13:13.000000000 +0100
@@ -91,14 +91,15 @@
       b.append('\n');
       int charno = error.getCharno();
 
-      // padding equal to the excerpt and arrow at the end
-      // charno == sourceExpert.length() means something is missing
-      // at the end of the line
-      if (excerpt.equals(LINE)
-          && 0 <= charno && charno < sourceExcerpt.length()) {
+ 
+ 
+ 
+			if (true) 
+			{
         for (int i = 0; i < charno; i++) {
           char c = sourceExcerpt.charAt(i);
           if (Character.isWhitespace(c)) {
+
             b.append(c);
           } else {
             b.append(' ');
