--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_44/CodeConsumer/Closure_44_CodeConsumer_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_44/CodeConsumer/Closure_44_CodeConsumer_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -191,11 +191,13 @@
       // need space to separate. This is not pretty printing.
       // For example: "return foo;"
       append(" ");
+    } else if (c == '/' && getLastChar() == '/') {
       // Do not allow a forward slash to appear after a DIV.
       // For example,
       // REGEXP DIV REGEXP
       // is valid and should print like
       // / // / /
+      append(" ");
     }
 
     append(newcode);
