--- /tmp/GenProg_Defects4J_Math_40/src/main/java/org/apache/commons/math/analysis/solvers/BaseAbstractUnivariateRealSolver.java	2019-08-01 01:58:03.152725754 -0400
+++ /tmp/GenProg_Defects4J_Math_40/patches_ezc5/Patch_726/patched/org/apache/commons/math/analysis/solvers/BaseAbstractUnivariateRealSolver.java	2019-08-01 03:20:42.986821790 -0400
@@ -293,7 +293,8 @@
         try {
             evaluations.incrementCount();
         } catch (MaxCountExceededException e) {
-            throw new TooManyEvaluationsException(e.getMax());
+            double initial = getStartValue();
+			throw new TooManyEvaluationsException(e.getMax());
         }
     }
 }
