--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Chart_8/Week/Chart_8_Week_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Chart_8/Week/Chart_8_Week_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -172,7 +172,7 @@
      */
     public Week(Date time, TimeZone zone) {
         // defer argument checking...
-        this(time, RegularTimePeriod.DEFAULT_TIME_ZONE, Locale.getDefault());
+        this(time, zone, Locale.getDefault());
     }
 
     /**
