--- /Users/haoye.tian/Documents/University/data/PatchCollecting/Cardumen/Incorrect/Chart/patch1-Chart-6-Cardumen.buggy	2020-11-02 19:13:13.000000000 +0100
+++ /Users/haoye.tian/Documents/University/data/PatchCollecting/Cardumen/Incorrect/Chart/patch1-Chart-6-Cardumen.fixed	2020-11-02 21:29:06.000000000 +0100
@@ -78,10 +78,11 @@
      * @param shape  the {@link Shape}.
      */
     public void setShape(int index, Shape shape) {
-        set(index, shape);
+		set(AbstractObjectList.DEFAULT_INITIAL_CAPACITY, shape);
     }
 
-    /**
+ 
+    /*
      * Returns an independent copy of the list.
      *
      * @return A clone.
