--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_91/CheckGlobalThis/Closure_91_CheckGlobalThis_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_91/CheckGlobalThis/Closure_91_CheckGlobalThis_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -112,6 +112,15 @@
       }
 
       // Don't traverse functions that are getting lent to a prototype.
+      Node gramps = parent.getParent();
+      if (NodeUtil.isObjectLitKey(parent, gramps)) {
+        JSDocInfo maybeLends = gramps.getJSDocInfo();
+        if (maybeLends != null &&
+            maybeLends.getLendsName() != null &&
+            maybeLends.getLendsName().endsWith(".prototype")) {
+          return false;
+        }
+      }
     }
 
     if (parent != null && parent.getType() == Token.ASSIGN) {
