--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Chart_16/DefaultIntervalCategoryDataset/Chart_16_DefaultIntervalCategoryDataset_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Chart_16/DefaultIntervalCategoryDataset/Chart_16_DefaultIntervalCategoryDataset_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -204,8 +204,8 @@
 
             }
             else {
-                this.seriesKeys = null;
-                this.categoryKeys = null;
+                this.seriesKeys = new Comparable[0];
+                this.categoryKeys = new Comparable[0];
             }
         }
 
@@ -335,7 +335,7 @@
         if (categoryKeys == null) {
             throw new IllegalArgumentException("Null 'categoryKeys' argument.");
         }
-        if (categoryKeys.length != this.startData[0].length) {
+        if (categoryKeys.length != getCategoryCount()) {
             throw new IllegalArgumentException(
                     "The number of categories does not match the data.");
         }
