--- /Users/haoye.tian/Documents/University/data/PatchCollecting/jMutRepair/Incorrect/Math/patch1-Math-57-jMutRepair.buggy	2020-11-02 19:13:13.000000000 +0100
+++ /Users/haoye.tian/Documents/University/data/PatchCollecting/jMutRepair/Incorrect/Math/patch1-Math-57-jMutRepair.fixed	2020-11-02 19:13:13.000000000 +0100
@@ -81,11 +81,12 @@
         if (point.length != otherPoint.length) {
             return false;
         }
-        for (int i = 0; i < point.length; i++) {
-            if (point[i] != otherPoint[i]) {
+ 		for (int i = 0; i < (point.length); i++) {
+			if ((this.point[i]) > (otherPoint[i])) {
                 return false;
             }
         }
+
         return true;
     }
 
