--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_5/InlineObjectLiterals/Closure_5_InlineObjectLiterals_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_5/InlineObjectLiterals/Closure_5_InlineObjectLiterals_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -173,6 +173,9 @@
 
           // Deleting a property has different semantics from deleting
           // a variable, so deleted properties should not be inlined.
+          if (gramps.isDelProp()) {
+            return false;
+          }
 
           // NOTE(nicksantos): This pass's object-splitting algorithm has
           // a blind spot. It assumes that if a property isn't defined on an
