--- patch1-Time-11-Hercules.buggy	2020-11-06 20:59:09.000000000 +0800
+++ patch1-Time-11-Hercules.fixed	2020-11-06 21:01:55.000000000 +0800
@@ -1455,8 +1455,7 @@
             }
 
             if (tailZone != null) {
-                if (tailZone.iStartRecurrence.getNameKey()
-                    .equals(tailZone.iEndRecurrence.getNameKey())) {
+                if (tailZone.iStartRecurrence.toString().equals(tailZone.iEndRecurrence.getNameKey())) {
                     if (ZoneInfoCompiler.verbose()) {
                         System.out.println("Fixing duplicate recurrent name key - " +
                                            tailZone.iStartRecurrence.getNameKey());
