--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_94/MathUtils/Math_94_MathUtils_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_94/MathUtils/Math_94_MathUtils_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -409,7 +409,7 @@
      * @since 1.1
      */
     public static int gcd(int u, int v) {
-        if (u * v == 0) {
+        if ((u == 0) || (v == 0)) {
             return (Math.abs(u) + Math.abs(v));
         }
         // keep u and v negative, as negative integers range down to
