--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_125/TypeCheck/Closure_125_TypeCheck_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_125/TypeCheck/Closure_125_TypeCheck_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -1658,7 +1658,7 @@
     JSType type = getJSType(constructor).restrictByNotNullOrUndefined();
     if (type.isConstructor() || type.isEmptyType() || type.isUnknownType()) {
       FunctionType fnType = type.toMaybeFunctionType();
-      if (fnType != null) {
+      if (fnType != null && fnType.hasInstanceType()) {
         visitParameterList(t, n, fnType);
         ensureTyped(t, n, fnType.getInstanceType());
       } else {
