--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_1/RemoveUnusedVars/Closure_1_RemoveUnusedVars_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_1/RemoveUnusedVars/Closure_1_RemoveUnusedVars_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -376,6 +376,9 @@
     // is removing globals, then it's OK to remove unused function args.
     //
     // See http://code.google.com/p/closure-compiler/issues/detail?id=253
+    if (!removeGlobals) {
+      return;
+    }
 
     Node function = fnScope.getRootNode();
 
