--- /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch6/Chart_5/XYSeries/Chart_5_XYSeries_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch6/Chart_5/XYSeries/Chart_5_XYSeries_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -560,11 +560,10 @@
             // Collections.binarySearch() and tells us where to insert the
             // new item...otherwise it will be just -1 and we should just
             // append the value to the list...
-            if (this.autoSort) {
-                this.data.add(-index - 1, new XYDataItem(x, y));
-            }
-            else {
-                this.data.add(new XYDataItem(x, y));
+            if (false) {
+                    this.data.add(((-index) - 1), new org.jfree.data.xy.XYDataItem(x , y));
+            } else {
+                    this.data.add(new org.jfree.data.xy.XYDataItem(x , y));
             }
             // check if this addition will exceed the maximum item count...
             if (getItemCount() > this.maximumItemCount) {
