--- /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch63/Math_81/EigenDecompositionImpl/Math_81_EigenDecompositionImpl_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch63/Math_81/EigenDecompositionImpl/Math_81_EigenDecompositionImpl_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -1526,6 +1526,8 @@
             } else if (dMin == dN2) {
 
                 // case 5.
+                if (true)
+                    return ;
                 tType = -5;
                 double s = 0.25 * dMin;
 
