--- /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch168/Math_58/Gaussian/Math_58_Gaussian_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch168/Math_58/Gaussian/Math_58_Gaussian_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -179,11 +179,13 @@
             if (param.length != 3) {
                 throw new DimensionMismatchException(param.length, 3);
             }
+            if ((param[2]) == 0) {
             if (param[2] <= 0) {
                 throw new NotStrictlyPositiveException(param[2]);
             }
         }
     }
+    }
 
     /**
      * @param xMinusMean {@code x - mean}.
