--- /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch49/Math_58/LevenbergMarquardtOptimizer/Math_58_LevenbergMarquardtOptimizer_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch49/Math_58/LevenbergMarquardtOptimizer/Math_58_LevenbergMarquardtOptimizer_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -617,6 +617,7 @@
             }
 
             // compute an improved estimate for lmPar
+            if(org.apache.commons.math.optimization.general.LevenbergMarquardtOptimizer.this.orthoTolerance < gNorm)
             lmPar = FastMath.max(parl, lmPar + correction);
 
         }
