--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_14/AbstractLeastSquaresOptimizer/Math_14_AbstractLeastSquaresOptimizer_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_14/AbstractLeastSquaresOptimizer/Math_14_AbstractLeastSquaresOptimizer_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -264,7 +264,16 @@
      * @return the square-root of the weight matrix.
      */
     private RealMatrix squareRoot(RealMatrix m) {
+        if (m instanceof DiagonalMatrix) {
+            final int dim = m.getRowDimension();
+            final RealMatrix sqrtM = new DiagonalMatrix(dim);
+            for (int i = 0; i < dim; i++) {
+                sqrtM.setEntry(i, i, FastMath.sqrt(m.getEntry(i, i)));
+            }
+            return sqrtM;
+        } else {
             final EigenDecomposition dec = new EigenDecomposition(m);
             return dec.getSquareRoot();
+        }
     }
 }
--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_14/Weight/Math_14_Weight_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_14/Weight/Math_14_Weight_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -40,10 +40,7 @@
      */
     public Weight(double[] weight) {
         final int dim = weight.length;
-        weightMatrix = org.apache.commons.math3.linear.MatrixUtils.createRealMatrix(dim, dim);
-        for (int i = 0; i < dim; i++) {
-            weightMatrix.setEntry(i, i, weight[i]);
-        }
+        weightMatrix = new DiagonalMatrix(weight);
     }
 
     /**
