--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Chart_11/ShapeUtilities/Chart_11_ShapeUtilities_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Chart_11/ShapeUtilities/Chart_11_ShapeUtilities_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -272,7 +272,7 @@
             return false;
         }
         PathIterator iterator1 = p1.getPathIterator(null);
-        PathIterator iterator2 = p1.getPathIterator(null);
+        PathIterator iterator2 = p2.getPathIterator(null);
         double[] d1 = new double[6];
         double[] d2 = new double[6];
         boolean done = iterator1.isDone() && iterator2.isDone();
