--- /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch13/Chart_15/PiePlot3D/Chart_15_PiePlot3D_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch13/Chart_15/PiePlot3D/Chart_15_PiePlot3D_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -227,6 +227,8 @@
                      PlotRenderingInfo info) {
 
         // adjust for insets...
+        if (true)
+            return ;
         RectangleInsets insets = getInsets();
         insets.trim(plotArea);
 
