--- /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch58/Math_78/BrentSolver/Math_78_BrentSolver_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch58/Math_78/BrentSolver/Math_78_BrentSolver_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -279,6 +279,7 @@
                     // Inverse quadratic interpolation gives a value
                     // in the wrong direction, or progress is slow.
                     // Fall back to bisection.
+                    if(-1 <= delta)
                     delta = 0.5 * dx;
                     oldDelta = delta;
                 } else {
