--- /Users/haoye.tian/Documents/University/data/PatchCollecting/jKali/Incorrect/Closure/patch1-Closure-101-jKali.buggy	2020-11-02 19:13:13.000000000 +0100
+++ /Users/haoye.tian/Documents/University/data/PatchCollecting/jKali/Incorrect/Closure/patch1-Closure-101-jKali.fixed	2020-11-02 19:13:13.000000000 +0100
@@ -89,9 +89,9 @@
    * @param options The CompilerOptions object to set the options on.
    */
   private static void applySafeCompilationOptions(CompilerOptions options) {
-    // Does not call applyBasicCompilationOptions(options) because the call to
-    // skipAllCompilerPasses() cannot be easily undone.
-    options.closurePass = true;
+ 
+ 
+
     options.variableRenaming = VariableRenamingPolicy.LOCAL;
     options.inlineLocalVariables = true;
     options.checkGlobalThisLevel = CheckLevel.OFF;
