--- TimeSeries.java	2020-10-11 16:12:10.000000000 +0200
+++ TimeSeries copy.java	2020-11-05 20:55:59.000000000 +0100
@@ -941,7 +941,7 @@
             endIndex = -(endIndex + 1); // this is first item AFTER end period
             endIndex = endIndex - 1;    // so this is last item BEFORE end
         }
-        if (endIndex < 0) {
+        if( endIndex <0|| endIndex < startIndex ) {
             emptyRange = true;
         }
         if (emptyRange) {
