--- /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch81/Time_4/ZeroIsMaxDateTimeField/Time_4_ZeroIsMaxDateTimeField_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch81/Time_4/ZeroIsMaxDateTimeField/Time_4_ZeroIsMaxDateTimeField_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -175,6 +175,8 @@
      * @return the maximum value
      */
     public int getMaximumValue(ReadablePartial instant, int[] values) {
+        if (true)
+            return 0;
         return getWrappedField().getMaximumValue(instant, values) + 1;
     }
 
