--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Chart_19/CategoryPlot/Chart_19_CategoryPlot_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Chart_19/CategoryPlot/Chart_19_CategoryPlot_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -695,6 +695,9 @@
      * @since 1.0.3
      */
     public int getDomainAxisIndex(CategoryAxis axis) {
+        if (axis == null) {
+            throw new IllegalArgumentException("Null 'axis' argument.");
+        }
         return this.domainAxes.indexOf(axis);
     }
     
@@ -970,6 +973,9 @@
      * @since 1.0.7
      */
     public int getRangeAxisIndex(ValueAxis axis) {
+        if (axis == null) {
+            throw new IllegalArgumentException("Null 'axis' argument.");
+        }
         int result = this.rangeAxes.indexOf(axis);
         if (result < 0) { // try the parent plot
             Plot parent = getParent();
