--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_61/NodeUtil/Closure_61_NodeUtil_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_61/NodeUtil/Closure_61_NodeUtil_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -955,6 +955,12 @@
       }
 
       // Functions in the "Math" namespace have no side effects.
+      if (nameNode.getFirstChild().getType() == Token.NAME) {
+        String namespaceName = nameNode.getFirstChild().getString();
+        if (namespaceName.equals("Math")) {
+          return false;
+        }
+      }
 
       if (compiler != null && !compiler.hasRegExpGlobalReferences()) {
         if (nameNode.getFirstChild().getType() == Token.REGEXP
