--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_12/BitsStreamGenerator/Math_12_BitsStreamGenerator_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_12/BitsStreamGenerator/Math_12_BitsStreamGenerator_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -16,6 +16,7 @@
  */
 package org.apache.commons.math3.random;
 
+import java.io.Serializable;
 
 import org.apache.commons.math3.exception.NotStrictlyPositiveException;
 import org.apache.commons.math3.util.FastMath;
@@ -26,8 +27,10 @@
  * @since 2.0
  */
 public abstract class BitsStreamGenerator
-    implements RandomGenerator {
+    implements RandomGenerator,
+               Serializable {
     /** Serializable version identifier */
+    private static final long serialVersionUID = 20130104L;
     /** Next gaussian. */
     private double nextGaussian;
 
