--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_109/JsDocInfoParser/Closure_109_JsDocInfoParser_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_109/JsDocInfoParser/Closure_109_JsDocInfoParser_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -1905,7 +1905,11 @@
    * For expressions on the right hand side of a this: or new:
    */
   private Node parseContextTypeExpression(JsDocToken token) {
-          return parseTypeName(token);
+    if (token == JsDocToken.QMARK) {
+      return newNode(Token.QMARK);
+    } else {
+      return parseBasicTypeExpression(token);
+    }
   }
 
   /**
