--- /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch150/Lang_44/NumberUtils/Lang_44_NumberUtils_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch150/Lang_44/NumberUtils/Lang_44_NumberUtils_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -190,9 +190,7 @@
             switch (lastChar) {
                 case 'l' :
                 case 'L' :
-                    if (dec == null
-                        && exp == null
-                        && (numeric.charAt(0) == '-' && isDigits(numeric.substring(1)) || isDigits(numeric))) {
+                    if (((dec == null) && (exp == null))) {
                         try {
                             return createLong(numeric);
                         } catch (NumberFormatException nfe) {
