--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_129/PrepareAst/Closure_129_PrepareAst_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_129/PrepareAst/Closure_129_PrepareAst_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -163,6 +163,9 @@
       Node first = n.getFirstChild();
 
       // ignore cast nodes.
+      while (first.isCast()) {
+        first = first.getFirstChild();
+      }
 
       if (!NodeUtil.isGet(first)) {
         n.putBooleanProp(Node.FREE_CALL, true);
