--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_11/TypeCheck/Closure_11_TypeCheck_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_11/TypeCheck/Closure_11_TypeCheck_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -1311,8 +1311,6 @@
 
     if (childType.isDict()) {
       report(t, property, TypeValidator.ILLEGAL_PROPERTY_ACCESS, "'.'", "dict");
-    } else if (n.getJSType() != null && parent.isAssign()) {
-      return;
     } else if (validator.expectNotNullOrUndefined(t, n, childType,
         "No properties on this expression", getNativeType(OBJECT_TYPE))) {
       checkPropertyAccess(childType, property.getString(), t, n);
