--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_24/BrentOptimizer/Math_24_BrentOptimizer_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_24/BrentOptimizer/Math_24_BrentOptimizer_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -227,7 +227,7 @@
 
                 if (checker != null) {
                     if (checker.converged(iter, previous, current)) {
-                        return current;
+                        return best(current, previous, isMinim);
                     }
                 }
 
@@ -264,7 +264,7 @@
                     }
                 }
             } else { // Default termination (Brent's criterion).
-                return current;
+                return best(current, previous, isMinim);
             }
             ++iter;
         }
