--- LocaleUtils.java	2020-10-10 17:42:30.000000000 +0200
+++ LocaleUtils copy.java	2020-11-07 15:23:12.000000000 +0100
@@ -220,7 +220,7 @@
      * @return true if the locale is a known locale
      */
     public static boolean isAvailableLocale(Locale locale) {
-        return cAvailableLocaleSet.contains(locale);
+        return cAvailableLocaleList.contains(locale);
     }
 
     //-----------------------------------------------------------------------
