--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_107/CommandLineRunner/Closure_107_CommandLineRunner_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_107/CommandLineRunner/Closure_107_CommandLineRunner_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -859,6 +859,7 @@
       // so we might as well inline it. But shut off the i18n warnings,
       // because the user didn't really ask for i18n.
       options.messageBundle = new EmptyMessageBundle();
+      options.setWarningLevel(JsMessageVisitor.MSG_CONVENTIONS, CheckLevel.OFF);
     }
 
     return options;
