--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_87/SimplexTableau/Math_87_SimplexTableau_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_87/SimplexTableau/Math_87_SimplexTableau_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -272,12 +272,10 @@
     private Integer getBasicRow(final int col) {
         Integer row = null;
         for (int i = getNumObjectiveFunctions(); i < getHeight(); i++) {
-            if (!MathUtils.equals(getEntry(i, col), 0.0, epsilon)) {
-                if (row == null) {
+            if (MathUtils.equals(getEntry(i, col), 1.0, epsilon) && (row == null)) {
                 row = i;
-                } else {
+            } else if (!MathUtils.equals(getEntry(i, col), 0.0, epsilon)) {
                 return null;
-                }
             }
         }
         return row;
