--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Lang_18/FastDateFormat/Lang_18_FastDateFormat_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Lang_18/FastDateFormat/Lang_18_FastDateFormat_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -492,10 +492,10 @@
                 rule = new TextField(Calendar.ERA, ERAs);
                 break;
             case 'y': // year (number)
-                if (tokenLen >= 4) {
-                    rule = selectNumberRule(Calendar.YEAR, tokenLen);
-                } else {
+                if (tokenLen == 2) {
                     rule = TwoDigitYearField.INSTANCE;
+                } else {
+                    rule = selectNumberRule(Calendar.YEAR, tokenLen < 4 ? 4 : tokenLen);
                 }
                 break;
             case 'M': // month in year (text and number)
