--- MannWhitneyUTest.java	2020-10-10 17:45:29.000000000 +0200
+++ MannWhitneyUTest copy.java	2020-11-05 22:44:31.000000000 +0100
@@ -170,7 +170,7 @@
                                              final int n2)
         throws ConvergenceException, MaxCountExceededException {
 
-        final int n1n2prod = n1 * n2;
+        final long n1n2prod = n1*n2;
 
         // http://en.wikipedia.org/wiki/Mann%E2%80%93Whitney_U#Normal_approximation
         final double EU = n1n2prod / 2.0;
