--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_105/SimpleRegression/Math_105_SimpleRegression_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_105/SimpleRegression/Math_105_SimpleRegression_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -261,7 +261,7 @@
      * @return sum of squared errors associated with the regression model
      */
     public double getSumSquaredErrors() {
-        return sumYY - sumXY * sumXY / sumXX;
+        return Math.max(0d, sumYY - sumXY * sumXY / sumXX);
     }
 
     /**
