--- HypergeometricDistribution.java	2020-10-10 17:43:24.000000000 +0200
+++ HypergeometricDistribution copy.java	2020-11-05 22:35:26.000000000 +0100
@@ -265,8 +265,7 @@
      * size {@code n}, the mean is {@code n * m / N}.
      */
     public double getNumericalMean() {
-        return (double) (getSampleSize() * getNumberOfSuccesses()) / (double) getPopulationSize();
-    }
+        return( double ) getSampleSize( ) *getNumberOfSuccesses( ) /( double ) getPopulationSize( ) ;    }
 
     /**
      * {@inheritDoc}
