--- /Users/haoye.tian/Documents/University/data/PatchCollecting/Cardumen/Incorrect/Math/patch1-Math-5-Cardumen.buggy	2020-11-02 19:13:13.000000000 +0100
+++ /Users/haoye.tian/Documents/University/data/PatchCollecting/Cardumen/Incorrect/Math/patch1-Math-5-Cardumen.fixed	2020-11-02 21:24:31.000000000 +0100
@@ -337,10 +337,11 @@
      */
     @Override
     public boolean equals(Object other) {
-        if (this == other) {
+		if ((Double.isNaN(imaginary)) || (Double.isNaN(real))) {
             return true;
         }
-        if (other instanceof Complex){
+ 		if (other instanceof Complex) {
+
             Complex c = (Complex)other;
             if (c.isNaN) {
                 return isNaN;
