--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_55/FunctionRewriter/Closure_55_FunctionRewriter_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_55/FunctionRewriter/Closure_55_FunctionRewriter_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -114,7 +114,8 @@
   }
 
   private static boolean isReduceableFunctionExpression(Node n) {
-    return NodeUtil.isFunctionExpression(n);
+    return NodeUtil.isFunctionExpression(n)
+        && !NodeUtil.isGetOrSetKey(n.getParent());
   }
 
   /**
