--- /Users/haoye.tian/Documents/University/data/PatchCollecting/jMutRepair/Incorrect/Math/patch1-Math-88-jMutRepair.buggy	2020-11-02 19:13:13.000000000 +0100
+++ /Users/haoye.tian/Documents/University/data/PatchCollecting/jMutRepair/Incorrect/Math/patch1-Math-88-jMutRepair.fixed	2020-11-02 19:13:13.000000000 +0100
@@ -77,13 +77,14 @@
         double minRatio = Double.MAX_VALUE;
         Integer minRatioPos = null;
         for (int i = tableau.getNumObjectiveFunctions(); i < tableau.getHeight(); i++) {
-            double rhs = tableau.getEntry(i, tableau.getWidth() - 1);
-            if (MathUtils.compareTo(tableau.getEntry(i, col), 0, epsilon) >= 0) {
-                double ratio = rhs / tableau.getEntry(i, col);
-                if (ratio < minRatio) {
+ 			double rhs = tableau.getEntry(i, ((tableau.getWidth()) - 1));
+ 			if ((MathUtils.compareTo(tableau.getEntry(i, col), 0, epsilon)) >= 0) {
+ 				double ratio = rhs / (tableau.getEntry(i, col));
+				if (ratio <= minRatio) {
                     minRatio = ratio;
                     minRatioPos = i; 
                 }
+
             }
         }
         return minRatioPos;
