--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Time_25/DateTimeZone/Time_25_DateTimeZone_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Time_25/DateTimeZone/Time_25_DateTimeZone_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -896,6 +896,15 @@
                     return offsetLocal;
                 }
             }
+        } else if (offsetLocal > 0) {
+            long prev = previousTransition(instantAdjusted);
+            if (prev < instantAdjusted) {
+                int offsetPrev = getOffset(prev);
+                int diff = offsetPrev - offsetLocal;
+                if (instantAdjusted - prev <= diff) {
+                    return offsetPrev;
+                }
+            }
         }
         return offsetAdjusted;
     }
