--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_53/Complex/Math_53_Complex_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_53/Complex/Math_53_Complex_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -150,6 +150,9 @@
     public Complex add(Complex rhs)
         throws NullArgumentException {
         MathUtils.checkNotNull(rhs);
+        if (isNaN || rhs.isNaN) {
+            return NaN;
+        }
         return createComplex(real + rhs.getReal(),
             imaginary + rhs.getImaginary());
     }
