--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_58/LiveVariablesAnalysis/Closure_58_LiveVariablesAnalysis_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_58/LiveVariablesAnalysis/Closure_58_LiveVariablesAnalysis_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -203,8 +203,12 @@
             // for(var x in y) {...}
             lhs = lhs.getLastChild();
           }
+          if (NodeUtil.isName(lhs)) {
             addToSetIfLocal(lhs, kill);
             addToSetIfLocal(lhs, gen);
+          } else {
+            computeGenKill(lhs, gen, kill, conditional);
+          }
           computeGenKill(rhs, gen, kill, conditional);
         }
         return;
