--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_34/CodeGenerator/Closure_34_CodeGenerator_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_34/CodeGenerator/Closure_34_CodeGenerator_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -119,9 +119,7 @@
         cc.addOp(opstr, true);
         addExpr(last, p, rhsContext);
       } else {
-        addExpr(first, p, context);
-        cc.addOp(opstr, true);
-        addExpr(last, p + 1, rhsContext);
+        unrollBinaryOperator(n, type, opstr, context, rhsContext, p, p + 1);
       }
       return;
     }
@@ -751,7 +749,7 @@
     Node current = firstNonOperator;
     do {
       current = current.getParent();
-      cc.listSeparator();
+      cc.addOp(opStr, true);
       addExpr(current.getFirstChild().getNext(), rightPrecedence, rhsContext);
     } while (current != n);
   }
--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_34/CodePrinter/Closure_34_CodePrinter_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_34/CodePrinter/Closure_34_CodePrinter_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -331,7 +331,7 @@
     @Override
     void appendOp(String op, boolean binOp) {
       if (binOp) {
-        if (getLastChar() != ' ') {
+        if (getLastChar() != ' ' && op.charAt(0) != ',') {
           append(" ");
         }
         append(op);
