--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_37/IRFactory/Closure_37_IRFactory_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_37/IRFactory/Closure_37_IRFactory_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -664,9 +664,13 @@
       node.addChildToBack(lp);
 
       Node bodyNode = transform(functionNode.getBody());
+      if (!bodyNode.isBlock()) {
         // When in ideMode Rhino tries to parse some constructs the compiler
         // doesn't support, repair it here. see Rhino's
         // Parser#parseFunctionBodyExpr.
+        Preconditions.checkState(config.isIdeMode);
+        bodyNode = IR.block();
+      }
       parseDirectives(bodyNode);
       node.addChildToBack(bodyNode);
      return node;
--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_37/NodeTraversal/Closure_37_NodeTraversal_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_37/NodeTraversal/Closure_37_NodeTraversal_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -538,7 +538,7 @@
 
     // Body
     Preconditions.checkState(body.getNext() == null &&
-            body.isBlock());
+            body.isBlock(), body);
     traverseBranch(body, n);
 
     popScope();
