--- /Users/haoye.tian/Documents/University/data/PatchCollecting/jKali/Incorrect/Closure/patch1-Closure-45-jKali.buggy	2020-11-02 19:13:13.000000000 +0100
+++ /Users/haoye.tian/Documents/University/data/PatchCollecting/jKali/Incorrect/Closure/patch1-Closure-45-jKali.fixed	2020-11-02 19:13:13.000000000 +0100
@@ -732,11 +732,12 @@
           for (Assign assign : assignsByVar.get(var)) {
             if (assign.isPropertyAssign) {
               hasPropertyAssign = true;
-            } else if (!NodeUtil.isLiteralValue(
-                assign.assignNode.getLastChild(), true)) {
+						}else { 							if (true) 
+							{
               assignedToUnknownValue = true;
+ 							}}
             }
-          }
+
 
           if (assignedToUnknownValue && hasPropertyAssign) {
             changes = markReferencedVar(var) || changes;
