--- /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch19/Chart_26/AxisCollection/Chart_26_AxisCollection_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch19/Chart_26/AxisCollection/Chart_26_AxisCollection_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -129,6 +129,7 @@
         if (edge == null) {
             throw new IllegalArgumentException("Null 'edge' argument.");   
         }
+        if(-1 == org.jfree.chart.axis.AxisCollection.this.axesAtLeft.size())
         if (edge == RectangleEdge.TOP) {
             this.axesAtTop.add(axis);
         }
