--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_50/BaseSecantSolver/Math_50_BaseSecantSolver_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_50/BaseSecantSolver/Math_50_BaseSecantSolver_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -184,10 +184,6 @@
                     break;
                 case REGULA_FALSI:
                     // Nothing.
-                    if (x == x1) {
-                        x0 = 0.5 * (x0 + x1 - FastMath.max(rtol * FastMath.abs(x1), atol));
-                        f0 = computeObjectiveValue(x0);
-                    }
                     break;
                 default:
                     // Should never happen.
