--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_15/FlowSensitiveInlineVariables/Closure_15_FlowSensitiveInlineVariables_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_15/FlowSensitiveInlineVariables/Closure_15_FlowSensitiveInlineVariables_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -99,6 +99,9 @@
           return true;
         }
 
+        if (n.isDelProp()) {
+          return true;
+        }
 
         for (Node c = n.getFirstChild(); c != null; c = c.getNext()) {
           if (!ControlFlowGraph.isEnteringNewCfgNode(c) && apply(c)) {
