--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_70/TypedScopeCreator/Closure_70_TypedScopeCreator_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_70/TypedScopeCreator/Closure_70_TypedScopeCreator_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -1742,7 +1742,7 @@
           for (Node astParameter : astParameters.children()) {
             if (jsDocParameter != null) {
               defineSlot(astParameter, functionNode,
-                  jsDocParameter.getJSType(), true);
+                  jsDocParameter.getJSType(), false);
               jsDocParameter = jsDocParameter.getNext();
             } else {
               defineSlot(astParameter, functionNode, null, true);
