--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_62/LightweightMessageFormatter/Closure_62_LightweightMessageFormatter_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_62/LightweightMessageFormatter/Closure_62_LightweightMessageFormatter_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -95,7 +95,7 @@
       // charno == sourceExpert.length() means something is missing
       // at the end of the line
       if (excerpt.equals(LINE)
-          && 0 <= charno && charno < sourceExcerpt.length()) {
+          && 0 <= charno && charno <= sourceExcerpt.length()) {
         for (int i = 0; i < charno; i++) {
           char c = sourceExcerpt.charAt(i);
           if (Character.isWhitespace(c)) {
