--- /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch158/Math_28/SimplexSolver/Math_28_SimplexSolver_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch158/Math_28/SimplexSolver/Math_28_SimplexSolver_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -148,8 +148,10 @@
                         }
                     }
                 }
+                if (minRatioPositions.isEmpty()) {
                 return minRow;
         }
+        }
         return minRatioPositions.get(0);
     }
 
