--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Chart_20/ValueMarker/Chart_20_ValueMarker_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Chart_20/ValueMarker/Chart_20_ValueMarker_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -92,7 +92,7 @@
      */
     public ValueMarker(double value, Paint paint, Stroke stroke, 
                        Paint outlinePaint, Stroke outlineStroke, float alpha) {
-        super(paint, stroke, paint, stroke, alpha);
+        super(paint, stroke, outlinePaint, outlineStroke, alpha);
         this.value = value;
     }
     
