--- patch1-Time-4-Hercules.buggy	2020-11-06 20:26:24.000000000 +0800
+++ patch1-Time-4-Hercules.fixed	2020-11-06 20:27:08.000000000 +0800
@@ -461,7 +461,7 @@
             System.arraycopy(iValues, i, newValues, i + 1, newValues.length - i - 1);
             // use public constructor to ensure full validation
             // this isn't overly efficient, but is safe
-            Partial newPartial = new Partial(iChronology, newTypes, newValues);
+            Partial newPartial = new org.joda.time.Partial(newTypes , newValues , iChronology);
             iChronology.validate(newPartial, newValues);
             return newPartial;
         }
