--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Lang_38/FastDateFormat/Lang_38_FastDateFormat_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Lang_38/FastDateFormat/Lang_38_FastDateFormat_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -869,6 +869,7 @@
      */
     public StringBuffer format(Calendar calendar, StringBuffer buf) {
         if (mTimeZoneForced) {
+            calendar.getTime(); /// LANG-538
             calendar = (Calendar) calendar.clone();
             calendar.setTimeZone(mTimeZone);
         }
