--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_90/FunctionType/Closure_90_FunctionType_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_90/FunctionType/Closure_90_FunctionType_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -877,6 +877,9 @@
     // mean "nullable Foo". For certain tags (like @extends) we de-nullify
     // the name for them.
     JSType maybeTypeOfThis = safeResolve(typeOfThis, t, scope);
+    if (maybeTypeOfThis != null) {
+      maybeTypeOfThis = maybeTypeOfThis.restrictByNotNullOrUndefined();
+    }
     if (maybeTypeOfThis instanceof ObjectType) {
       typeOfThis = (ObjectType) maybeTypeOfThis;
     }
--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_90/FunctionTypeBuilder/Closure_90_FunctionTypeBuilder_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_90/FunctionTypeBuilder/Closure_90_FunctionTypeBuilder_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -181,7 +181,7 @@
       // create interfaces JSType, ObjectType, FunctionType etc and have
       // separate implementation instead of the class hierarchy, so that
       // union types can also be object types, etc.
-      if (!type.isSubtype(
+      if (!type.restrictByNotNullOrUndefined().isSubtype(
               typeRegistry.getNativeType(OBJECT_TYPE))) {
         reportWarning(THIS_TYPE_NON_OBJECT, type.toString());
         return false;
