--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Chart_3/TimeSeries/Chart_3_TimeSeries_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Chart_3/TimeSeries/Chart_3_TimeSeries_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -1054,6 +1054,8 @@
             throw new IllegalArgumentException("Requires start <= end.");
         }
         TimeSeries copy = (TimeSeries) super.clone();
+        copy.minY = Double.NaN;
+        copy.maxY = Double.NaN;
         copy.data = new java.util.ArrayList();
         if (this.data.size() > 0) {
             for (int index = start; index <= end; index++) {
