--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_89/Frequency/Math_89_Frequency_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_89/Frequency/Math_89_Frequency_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -107,7 +107,11 @@
      */
     @Deprecated
     public void addValue(Object v) {
+        if (v instanceof Comparable<?>){
             addValue((Comparable<?>) v);            
+        } else {
+            throw new IllegalArgumentException("Object must implement Comparable");
+        }
     }
     
     /**
