--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_53/InlineObjectLiterals/Closure_53_InlineObjectLiterals_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_53/InlineObjectLiterals/Closure_53_InlineObjectLiterals_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -328,6 +328,9 @@
       }
 
       Node replacement;
+      if (nodes.isEmpty()) {
+        replacement = new Node(Token.TRUE);
+      } else {
         // All assignments evaluate to true, so make sure that the
         // expr statement evaluates to true in case it matters.
         nodes.add(new Node(Token.TRUE));
@@ -347,6 +350,7 @@
         }
         cur.addChildToFront(nodes.get(i));
         cur.addChildToFront(nodes.get(i + 1));
+      }
 
       Node replace = ref.getParent();
       replacement.copyInformationFromForTree(replace);
