--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_67/MultiStartUnivariateRealOptimizer/Math_67_MultiStartUnivariateRealOptimizer_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Math_67/MultiStartUnivariateRealOptimizer/Math_67_MultiStartUnivariateRealOptimizer_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -89,12 +89,12 @@
 
     /** {@inheritDoc} */
     public double getFunctionValue() {
-        return optimizer.getFunctionValue();
+        return optimaValues[0];
     }
 
     /** {@inheritDoc} */
     public double getResult() {
-        return optimizer.getResult();
+        return optima[0];
     }
 
     /** {@inheritDoc} */
