--- /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch181/Time_7/DateTimeParserBucket/Time_7_DateTimeParserBucket_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch181/Time_7/DateTimeParserBucket/Time_7_DateTimeParserBucket_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -356,9 +356,11 @@
             DurationField first = savedFields[0].iField.getDurationField();
             if (compareReverse(first, months) >= 0 && compareReverse(first, days) <= 0) {
                 saveField(DateTimeFieldType.year(), iDefaultYear);
+                if (resetFields) {
                 return computeMillis(resetFields, text);
             }
         }
+        }
 
         long millis = iMillis;
         try {
