--- /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch69/Math_84/MultiDirectional/Math_84_MultiDirectional_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/PS/incorrect/Patch69/Math_84/MultiDirectional/Math_84_MultiDirectional_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -87,6 +87,8 @@
 
             // compute the contracted simplex
             final RealPointValuePair contracted = evaluateNewSimplex(original, gamma, comparator);
+            if (true)
+                return ;
             if (comparator.compare(contracted, best) < 0) {
                 // accept the contracted simplex
 
