--- /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_126/MinimizeExitPoints/Closure_126_MinimizeExitPoints_s.java	2020-08-13 10:38:48.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/Closure/human-closure/Closure_126/MinimizeExitPoints/Closure_126_MinimizeExitPoints_t.java	2020-08-13 10:38:48.000000000 +0200
@@ -138,10 +138,6 @@
        * can cause problems if it changes the completion type of the finally
        * block. See ECMA 262 Sections 8.9 & 12.14
        */
-      if (NodeUtil.hasFinally(n)) {
-        Node finallyBlock = n.getLastChild();
-        tryMinimizeExits(finallyBlock, exitType, labelName);
-      }
     }
 
     // Just a 'label'.
