--- patch1-Math-50-Hercules.buggy	2020-11-06 20:52:04.000000000 +0800
+++ patch1-Math-50-Hercules.fixed	2020-11-06 20:53:01.000000000 +0800
@@ -185,7 +185,7 @@
                 case REGULA_FALSI:
                     // Nothing.
                     if (x == x1) {
-                        x0 = 0.5 * (x0 + x1 - FastMath.max(rtol * FastMath.abs(x1), atol));
+                        x0 = 0.5 * ((x0 + x1) - (FastMath.max((rtol * (FastMath.asin(x1))), atol)));
                         f0 = computeObjectiveValue(x0);
                     }
                     break;
