--- a/src/main/java/org/jsoup/nodes/DocumentType.java
+++ b/src/main/java/org/jsoup/nodes/DocumentType.java
@@ -1,6 +1,7 @@
 package org.jsoup.nodes;
 
 import org.jsoup.helper.StringUtil;
+import org.jsoup.helper.Validate;
 
 /**
  * A {@code <!DOCTPYE>} node.
@@ -18,6 +19,7 @@ public class DocumentType extends Node {
     public DocumentType(String name, String publicId, String systemId, String baseUri) {
         super(baseUri);
 
+        Validate.notEmpty(name);
         attr("name", name);
         attr("publicId", publicId);
         attr("systemId", systemId);
@@ -30,11 +32,11 @@ public class DocumentType extends Node {
 
     @Override
     void outerHtmlHead(StringBuilder accum, int depth, Document.OutputSettings out) {
-        accum.append("<!DOCTYPE html");
+        accum.append("<!DOCTYPE ").append(attr("name"));
         if (!StringUtil.isBlank(attr("publicId")))
             accum.append(" PUBLIC \"").append(attr("publicId")).append("\"");
         if (!StringUtil.isBlank(attr("systemId")))
-            accum.append(' ').append(attr("systemId")).append("\"");
+            accum.append(" \"").append(attr("systemId")).append("\"");
         accum.append('>');
     }
 
