--- /Users/haoye.tian/Downloads/ODS/data/human_patches/Time_11/ZoneInfoCompiler/Time_11_ZoneInfoCompiler_s.java	2020-08-13 10:38:58.000000000 +0200
+++ /Users/haoye.tian/Downloads/ODS/data/human_patches/Time_11/ZoneInfoCompiler/Time_11_ZoneInfoCompiler_t.java	2020-08-13 10:38:58.000000000 +0200
@@ -65,10 +65,11 @@
 
     static Chronology cLenientISO;
 
-    static ThreadLocal<Boolean> cVerbose = new ThreadLocal<Boolean>();
-    static {
-        cVerbose.set(Boolean.FALSE);
-    }
+    static ThreadLocal<Boolean> cVerbose = new ThreadLocal<Boolean>() {
+        protected Boolean initialValue() {
+            return Boolean.FALSE;
+        }
+    };
 
     /**
      * Gets a flag indicating that verbose logging is required.
