--- GenericMetadataSupport.java	2020-10-10 17:51:13.000000000 +0200
+++ GenericMetadataSupport copy.java	2020-11-05 23:52:48.000000000 +0100
@@ -76,8 +76,8 @@
 
             if (actualTypeArgument instanceof WildcardType) {
                 contextualActualTypeParameters.put(typeParameter, boundsOf((WildcardType) actualTypeArgument));
-            } else {
-                contextualActualTypeParameters.put(typeParameter, actualTypeArgument);
+             } else if( typeParameter != actualTypeArgument ) { 
+                contextualActualTypeParameters.put( typeParameter , actualTypeArgument ) ;
             }
             // logger.log("For '" + parameterizedType + "' found type variable : { '" + typeParameter + "(" + System.identityHashCode(typeParameter) + ")" + "' : '" + actualTypeArgument + "(" + System.identityHashCode(typeParameter) + ")" + "' }");
         }
