--- a/src/main/java/org/jsoup/nodes/DocumentType.java
+++ b/src/main/java/org/jsoup/nodes/DocumentType.java
@@ -12,6 +12,7 @@ public class DocumentType extends Node {
     public static final String PUBLIC_KEY = "PUBLIC";
     public static final String SYSTEM_KEY = "SYSTEM";
     private static final String NAME = "name";
+    private static final String PUB_SYS_KEY = "pubSysKey"; // PUBLIC or SYSTEM
     private static final String PUBLIC_ID = "publicId";
     private static final String SYSTEM_ID = "systemId";
     // todo: quirk mode from publicId and systemId
@@ -28,6 +29,9 @@ public class DocumentType extends Node {
 
         attr(NAME, name);
         attr(PUBLIC_ID, publicId);
+        if (has(PUBLIC_ID)) {
+            attr(PUB_SYS_KEY, PUBLIC_KEY);
+        }
         attr(SYSTEM_ID, systemId);
     }
 
@@ -38,7 +42,16 @@ public class DocumentType extends Node {
      * @param systemId the doctype's system ID
      * @param baseUri the doctype's base URI
      */
+    public DocumentType(String name, String pubSysKey, String publicId, String systemId, String baseUri) {
+        super(baseUri);
 
+        attr(NAME, name);
+        if (pubSysKey != null) {
+            attr(PUB_SYS_KEY, pubSysKey);
+        }
+        attr(PUBLIC_ID, publicId);
+        attr(SYSTEM_ID, systemId);
+    }
 
     @Override
     public String nodeName() {
@@ -55,8 +68,10 @@ public class DocumentType extends Node {
         }
         if (has(NAME))
             accum.append(" ").append(attr(NAME));
+        if (has(PUB_SYS_KEY))
+            accum.append(" ").append(attr(PUB_SYS_KEY));
         if (has(PUBLIC_ID))
-            accum.append(" PUBLIC \"").append(attr(PUBLIC_ID)).append('"');
+            accum.append(" \"").append(attr(PUBLIC_ID)).append('"');
         if (has(SYSTEM_ID))
             accum.append(" \"").append(attr(SYSTEM_ID)).append('"');
         accum.append('>');
