--- a/src/main/java/org/jsoup/parser/Token.java
+++ b/src/main/java/org/jsoup/parser/Token.java
@@ -32,6 +32,7 @@ abstract class Token {
 
     static final class Doctype extends Token {
         final StringBuilder name = new StringBuilder();
+        String pubSysKey = null;
         final StringBuilder publicIdentifier = new StringBuilder();
         final StringBuilder systemIdentifier = new StringBuilder();
         boolean forceQuirks = false;
@@ -43,6 +44,7 @@ abstract class Token {
         @Override
         Token reset() {
             reset(name);
+            pubSysKey = null;
             reset(publicIdentifier);
             reset(systemIdentifier);
             forceQuirks = false;
@@ -53,6 +55,9 @@ abstract class Token {
             return name.toString();
         }
 
+        String getPubSysKey() {
+            return pubSysKey;
+        }
 
         String getPublicIdentifier() {
             return publicIdentifier.toString();
